From 39996bbc632702cc3182a1adce57517ec76432f0 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Thu, 23 Jul 2015 00:06:25 +0800
Subject: [PATCH 0959/1221] MLK-11276-1 ARM: dts: clean up dtb for imx7d

* Sort dtb nodes by alphabetical order;

* Some board level dtb file's iomux has two
  nodes, merge them and sort modules'
  pin setting by alphabetical order.

* Follow the same rule of using platform name
  for pin group instead of using module name.

Signed-off-by: Anson Huang <b20788@freescale.com>
[zou:Original patch taken from
git.freescale.com/imx/fsl-arm-yocto-bsp.git-b imx-3.14.52-1.1.0_ga]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 arch/arm/boot/dts/imx7d-12x12-ddr3-arm2.dts        |  523 ++++++------
 .../arm/boot/dts/imx7d-12x12-lpddr3-arm2-ecspi.dts |   16 +-
 .../boot/dts/imx7d-12x12-lpddr3-arm2-flexcan.dts   |    8 +-
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts   |   22 +-
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-mqs.dts  |    8 +-
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-sai.dts  |    8 +-
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts      |  881 +++++++++-----------
 arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-csi.dts    |    8 +-
 arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-mipi.dts   |   34 +-
 arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-sai.dts    |    8 +-
 arch/arm/boot/dts/imx7d-19x19-ddr3-arm2.dts        |  648 +++++++--------
 arch/arm/boot/dts/imx7d-19x19-lpddr3-arm2.dts      |  464 +++++------
 arch/arm/boot/dts/imx7d-sdb-epdc.dts               |    8 +-
 arch/arm/boot/dts/imx7d-sdb-gpmi-weim.dts          |    8 +-
 arch/arm/boot/dts/imx7d-sdb-m4.dts                 |   28 +-
 arch/arm/boot/dts/imx7d-sdb-touch.dts              |    7 +-
 arch/arm/boot/dts/imx7d-sdb.dts                    |  123 ++--
 17 files changed, 1324 insertions(+), 1478 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-12x12-ddr3-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-ddr3-arm2.dts
index 4f1e166..a5c97ca 100644
--- a/arch/arm/boot/dts/imx7d-12x12-ddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-ddr3-arm2.dts
@@ -134,6 +134,10 @@
 	};
 };
 
+&epxp {
+	status = "okay";
+};
+
 &flexcan1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_flexcan1>;
@@ -148,199 +152,122 @@
 	status = "disabled";
 };
 
-&iomuxc {
+&i2c1 {
+	clock-frequency = <100000>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hog_1>;
-
-	hog {
-		pinctrl_hog_1: hoggrp-1 {
-			fsl,pins = <
-				MX7D_PAD_GPIO1_IO14__GPIO1_IO14 	0x59
-				MX7D_PAD_GPIO1_IO15__GPIO1_IO15 	0x59
-				MX7D_PAD_SD2_RESET_B__GPIO5_IO11 	0x59
-			>;
-		};
-	};
+	pinctrl-0 = <&pinctrl_i2c1_1>;
+	status = "okay";
 
-	pwm1 {
-		pinctrl_pwm1: pwm1grp {
-			fsl,pins = <
-				MX7D_PAD_GPIO1_IO08__PWM1_OUT 0x110b0
-			>;
-		};
-	};
+	pmic: pfuze3000@08 {
+		compatible = "fsl,pfuze3000";
+		reg = <0x08>;
 
-	lcdif {
-		pinctrl_lcdif_dat: lcdifdatgrp {
-			fsl,pins = <
-				MX7D_PAD_EPDC_DATA00__LCD_DATA0  0x4001b0b0
-				MX7D_PAD_EPDC_DATA01__LCD_DATA1  0x4001b0b0
-				MX7D_PAD_EPDC_DATA02__LCD_DATA2  0x4001b0b0
-				MX7D_PAD_EPDC_DATA03__LCD_DATA3  0x4001b0b0
-				MX7D_PAD_EPDC_DATA04__LCD_DATA4  0x4001b0b0
-				MX7D_PAD_EPDC_DATA05__LCD_DATA5  0x4001b0b0
-				MX7D_PAD_EPDC_DATA06__LCD_DATA6  0x4001b0b0
-				MX7D_PAD_EPDC_DATA07__LCD_DATA7  0x4001b0b0
-				MX7D_PAD_EPDC_DATA08__LCD_DATA8  0x4001b0b0
-				MX7D_PAD_EPDC_DATA09__LCD_DATA9  0x4001b0b0
-				MX7D_PAD_EPDC_DATA10__LCD_DATA10 0x4001b0b0
-				MX7D_PAD_EPDC_DATA11__LCD_DATA11 0x4001b0b0
-				MX7D_PAD_EPDC_DATA12__LCD_DATA12 0x4001b0b0
-				MX7D_PAD_EPDC_DATA13__LCD_DATA13 0x4001b0b0
-				MX7D_PAD_EPDC_DATA14__LCD_DATA14 0x4001b0b0
-				MX7D_PAD_EPDC_DATA15__LCD_DATA15 0x4001b0b0
-				MX7D_PAD_EPDC_SDLE__LCD_DATA16   0x4001b0b0
-				MX7D_PAD_EPDC_SDOE__LCD_DATA17   0x4001b0b0
-				MX7D_PAD_EPDC_SDSHR__LCD_DATA18  0x4001b0b0
-				MX7D_PAD_EPDC_SDCE0__LCD_DATA19  0x4001b0b0
-				MX7D_PAD_EPDC_SDCE1__LCD_DATA20  0x4001b0b0
-				MX7D_PAD_EPDC_SDCE2__LCD_DATA21  0x4001b0b0
-				MX7D_PAD_EPDC_SDCE3__LCD_DATA22  0x4001b0b0
-				MX7D_PAD_EPDC_GDCLK__LCD_DATA23  0x4001b0b0
-			>;
-		};
+		regulators {
+			sw1a_reg: sw1a {
+				regulator-min-microvolt = <700000>;
+				regulator-max-microvolt = <1475000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+			/* use sw1c_reg to align with pfuze100/pfuze200 */
+			sw1c_reg: sw1b {
+				regulator-min-microvolt = <700000>;
+				regulator-max-microvolt = <1475000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
 
-		pinctrl_lcdif_ctrl: lcdifctrlgrp {
-			fsl,pins = <
-				MX7D_PAD_EPDC_SDCLK__LCD_CLK 		0x4001b0b0
-				MX7D_PAD_EPDC_BDR1__LCD_ENABLE 		0x4001b0b0
-				MX7D_PAD_EPDC_PWR_STAT__LCD_VSYNC 	0x4001b0b0
-				MX7D_PAD_EPDC_PWR_COM__LCD_HSYNC 	0x4001b0b0
-			>;
-		};
-	};
-};
+			sw2_reg: sw2 {
+				regulator-min-microvolt = <1500000>;
+				regulator-max-microvolt = <1850000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
 
-&lcdif {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lcdif_dat
-		     &pinctrl_lcdif_ctrl>;
-	display = <&display0>;
-	status = "okay";
+			sw3a_reg: sw3 {
+				regulator-min-microvolt = <900000>;
+				regulator-max-microvolt = <1650000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
 
-	display0: display {
-		bits-per-pixel = <16>;
-		bus-width = <24>;
+			swbst_reg: swbst {
+				regulator-min-microvolt = <5000000>;
+				regulator-max-microvolt = <5150000>;
+			};
 
-		display-timings {
-			native-mode = <&timing0>;
-			timing0: timing0 {
-				clock-frequency = <33500000>;
-				hactive = <800>;
-				vactive = <480>;
-				hback-porch = <89>;
-				hfront-porch = <164>;
-				vback-porch = <23>;
-				vfront-porch = <10>;
-				hsync-len = <10>;
-				vsync-len = <10>;
-				hsync-active = <0>;
-				vsync-active = <0>;
-				de-active = <1>;
-				pixelclk-active = <0>;
+			snvs_reg: vsnvs {
+				regulator-min-microvolt = <1000000>;
+				regulator-max-microvolt = <3000000>;
+				regulator-boot-on;
+				regulator-always-on;
 			};
-		};
-	};
-};
 
-&pcie {
-	pinctrl-names = "default";
-	status = "disabled";
-};
+			vref_reg: vrefddr {
+				regulator-boot-on;
+				regulator-always-on;
+			};
 
-&pwm1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pwm1>;
-	status = "okay";
-};
+			vgen1_reg: vldo1 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
 
-&uart1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart1_1>;
-	status = "okay";
-};
+			vgen2_reg: vldo2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+			};
 
-&iomuxc_lpsr {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hog_2>;
+			vgen3_reg: vccsd {
+				regulator-min-microvolt = <2850000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
 
-	hog {
-		pinctrl_hog_2: hoggrp-2 {
-			fsl,pins = <
-				MX7D_PAD_GPIO1_IO02__GPIO1_IO2 	0x59 /* flexcan stby1 */
-				MX7D_PAD_GPIO1_IO03__GPIO1_IO3  0x59 /* flexcan stby2 */
-				MX7D_PAD_GPIO1_IO01__ANATOP_24M_OUT 0x80000000
-			>;
-		};
-	};
+			vgen4_reg: v33 {
+				regulator-min-microvolt = <2850000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
 
-	i2c1 {
-		pinctrl_i2c1_1: i2c1grp-1 {
-			fsl,pins = <
-				MX7D_PAD_GPIO1_IO04__I2C1_SCL 	0x4000007f
-				MX7D_PAD_GPIO1_IO05__I2C1_SDA   0x4000007f
-			>;
-		};
-	};
+			vgen5_reg: vldo3 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
 
-	i2c2 {
-		pinctrl_i2c2_1: i2c2grp-1 {
-			fsl,pins = <
-				MX7D_PAD_GPIO1_IO06__I2C2_SCL 	0x4000007f
-				MX7D_PAD_GPIO1_IO07__I2C2_SDA 	0x4000007f
-			>;
+			vgen6_reg: vldo4 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
 		};
 	};
 };
 
-&usbh {
-	status = "okay";
-};
-
-&usbotg1 {
-	vbus-supply = <&reg_usb_otg1_vbus>;
-	srp-disable;
-	hnp-disable;
-	adp-disable;
-	status = "okay";
-};
-
-&usbotg2 {
-	vbus-supply = <&reg_usb_otg2_vbus>;
-	srp-disable;
-	hnp-disable;
-	adp-disable;
-	status = "okay";
-};
-
-&usdhc2 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
-	pinctrl-0 = <&pinctrl_usdhc2_1>;
-	pinctrl-1 = <&pinctrl_usdhc2_1_100mhz>;
-	pinctrl-2 = <&pinctrl_usdhc2_1_200mhz>;
-	assigned-clocks = <&clks IMX7D_USDHC2_ROOT_CLK>;
-	assigned-clocks-rates = <400000000>;
-	bus-width = <8>;
-	tuning-step = <2>;
-	non-removable;
-	keep-power-in-suspend;
-	status = "okay";
-};
-
-&usdhc3 {
+&i2c3 {
+	clock-frequency = <100000>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_usdhc3_1>;
-	vmmc-supply = <&reg_sd3_vmmc>;
-	cd-gpios = <&gpio1 14>;
-	wp-gpios = <&gpio1 15>;
-	keep-power-in-suspend;
-	enable-sdio-wakeup;
-	no-1-8-v;
+	pinctrl-0 = <&pinctrl_i2c3_1>;
 	status = "okay";
 };
 
 &iomuxc {
-	ecspi4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog_1>;
+
+	imx7d-12x12-ddr3-arm2 {
+		pinctrl_hog_1: hoggrp-1 {
+			fsl,pins = <
+				MX7D_PAD_GPIO1_IO14__GPIO1_IO14 	0x59
+				MX7D_PAD_GPIO1_IO15__GPIO1_IO15 	0x59
+				MX7D_PAD_SD2_RESET_B__GPIO5_IO11 	0x59
+			>;
+		};
+
 		pinctrl_ecspi4_cs_1: ecspi4_cs_grp-1 {
 			fsl,pins = <
 				MX7D_PAD_SD1_CLK__GPIO5_IO3 	0x2
@@ -357,36 +284,28 @@
 				MX7D_PAD_SD1_CD_B__ECSPI4_MISO 	 	0x2
 			>;
 		};
-	};
 
-	flexcan1 {
 		pinctrl_flexcan1: flexcan1grp {
 			fsl,pins = <
 				MX7D_PAD_SD3_DATA5__FLEXCAN1_TX 	0x59
 				MX7D_PAD_SD3_DATA7__FLEXCAN1_RX 	0x59
 			>;
 		};
-	};
 
-	flexcan2 {
 		pinctrl_flexcan2: flexcan2grp {
 			fsl,pins = <
 				MX7D_PAD_SD3_DATA6__FLEXCAN2_TX 	0x59
 				MX7D_PAD_SD3_DATA4__FLEXCAN2_RX 	0x59
 			>;
 		};
-	};
 
-	gpio_keys {
 		pinctrl_gpio_keys: gpio_keysgrp {
 			fsl,pins = <
 				MX7D_PAD_LCD_DATA12__GPIO3_IO17 0x32
 				MX7D_PAD_LCD_DATA13__GPIO3_IO18 0x32
 			>;
 		};
-	};
 
-	i2c3 {
 		pinctrl_i2c3_1: i2c3grp-1 {
 			fsl,pins = <
 				MX7D_PAD_ENET1_RGMII_RD0__I2C3_SCL 0x4000007f
@@ -394,18 +313,57 @@
 			>;
 		};
 
-	};
-
-	i2c4 {
 		pinctrl_i2c4_1: i2c4grp-1 {
 			fsl,pins = <
 				MX7D_PAD_ENET1_RGMII_TD2__I2C4_SCL 0x4000007f
 				MX7D_PAD_ENET1_RGMII_TD3__I2C4_SDA 0x4000007f
 			>;
 		};
-	};
 
-	uart1 {
+		pinctrl_lcdif_dat: lcdifdatgrp {
+			fsl,pins = <
+				MX7D_PAD_EPDC_DATA00__LCD_DATA0  0x4001b0b0
+				MX7D_PAD_EPDC_DATA01__LCD_DATA1  0x4001b0b0
+				MX7D_PAD_EPDC_DATA02__LCD_DATA2  0x4001b0b0
+				MX7D_PAD_EPDC_DATA03__LCD_DATA3  0x4001b0b0
+				MX7D_PAD_EPDC_DATA04__LCD_DATA4  0x4001b0b0
+				MX7D_PAD_EPDC_DATA05__LCD_DATA5  0x4001b0b0
+				MX7D_PAD_EPDC_DATA06__LCD_DATA6  0x4001b0b0
+				MX7D_PAD_EPDC_DATA07__LCD_DATA7  0x4001b0b0
+				MX7D_PAD_EPDC_DATA08__LCD_DATA8  0x4001b0b0
+				MX7D_PAD_EPDC_DATA09__LCD_DATA9  0x4001b0b0
+				MX7D_PAD_EPDC_DATA10__LCD_DATA10 0x4001b0b0
+				MX7D_PAD_EPDC_DATA11__LCD_DATA11 0x4001b0b0
+				MX7D_PAD_EPDC_DATA12__LCD_DATA12 0x4001b0b0
+				MX7D_PAD_EPDC_DATA13__LCD_DATA13 0x4001b0b0
+				MX7D_PAD_EPDC_DATA14__LCD_DATA14 0x4001b0b0
+				MX7D_PAD_EPDC_DATA15__LCD_DATA15 0x4001b0b0
+				MX7D_PAD_EPDC_SDLE__LCD_DATA16   0x4001b0b0
+				MX7D_PAD_EPDC_SDOE__LCD_DATA17   0x4001b0b0
+				MX7D_PAD_EPDC_SDSHR__LCD_DATA18  0x4001b0b0
+				MX7D_PAD_EPDC_SDCE0__LCD_DATA19  0x4001b0b0
+				MX7D_PAD_EPDC_SDCE1__LCD_DATA20  0x4001b0b0
+				MX7D_PAD_EPDC_SDCE2__LCD_DATA21  0x4001b0b0
+				MX7D_PAD_EPDC_SDCE3__LCD_DATA22  0x4001b0b0
+				MX7D_PAD_EPDC_GDCLK__LCD_DATA23  0x4001b0b0
+			>;
+		};
+
+		pinctrl_lcdif_ctrl: lcdifctrlgrp {
+			fsl,pins = <
+				MX7D_PAD_EPDC_SDCLK__LCD_CLK 		0x4001b0b0
+				MX7D_PAD_EPDC_BDR1__LCD_ENABLE 		0x4001b0b0
+				MX7D_PAD_EPDC_PWR_STAT__LCD_VSYNC 	0x4001b0b0
+				MX7D_PAD_EPDC_PWR_COM__LCD_HSYNC 	0x4001b0b0
+			>;
+		};
+
+		pinctrl_pwm1: pwm1grp {
+			fsl,pins = <
+				MX7D_PAD_GPIO1_IO08__PWM1_OUT 0x110b0
+			>;
+		};
+
 		pinctrl_uart1_1: uart1grp-1 {
 			fsl,pins = <
 				MX7D_PAD_UART1_TX_DATA__UART1_DCE_TX 0x79
@@ -413,9 +371,6 @@
 			>;
 		};
 
-	};
-
-	usdhc2 {
 		pinctrl_usdhc2_1: usdhc2grp-1 {
 			fsl,pins = <
 				MX7D_PAD_SD2_CMD__SD2_CMD     	0x59
@@ -460,9 +415,7 @@
 				MX7D_PAD_ECSPI1_SS0__SD2_DATA7 	0x5b
 			>;
 		};
-	};
 
-	usdhc3 {
 		pinctrl_usdhc3_1: usdhc3grp-1 {
 			fsl,pins = <
 				MX7D_PAD_SD3_CMD__SD3_CMD	    0x59
@@ -473,117 +426,133 @@
 				MX7D_PAD_SD3_DATA3__SD3_DATA3   0x59
 			>;
 		};
-
 	};
 };
 
-&sdma {
-	status = "okay";
-};
+&iomuxc_lpsr {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog_2>;
 
-&epxp {
-	status = "okay";
+	imx7d-12x12-ddr3-arm2 {
+		pinctrl_hog_2: hoggrp-2 {
+			fsl,pins = <
+				MX7D_PAD_GPIO1_IO02__GPIO1_IO2 	0x59 /* flexcan stby1 */
+				MX7D_PAD_GPIO1_IO03__GPIO1_IO3  0x59 /* flexcan stby2 */
+				MX7D_PAD_GPIO1_IO01__ANATOP_24M_OUT 0x80000000
+			>;
+		};
+
+		pinctrl_i2c1_1: i2c1grp-1 {
+			fsl,pins = <
+				MX7D_PAD_GPIO1_IO04__I2C1_SCL 	0x4000007f
+				MX7D_PAD_GPIO1_IO05__I2C1_SDA   0x4000007f
+			>;
+		};
+
+		pinctrl_i2c2_1: i2c2grp-1 {
+			fsl,pins = <
+				MX7D_PAD_GPIO1_IO06__I2C2_SCL 	0x4000007f
+				MX7D_PAD_GPIO1_IO07__I2C2_SDA 	0x4000007f
+			>;
+		};
+	};
 };
 
-&i2c1 {
-	clock-frequency = <100000>;
+&lcdif {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c1_1>;
+	pinctrl-0 = <&pinctrl_lcdif_dat
+		     &pinctrl_lcdif_ctrl>;
+	display = <&display0>;
 	status = "okay";
 
-	pmic: pfuze3000@08 {
-		compatible = "fsl,pfuze3000";
-		reg = <0x08>;
-
-		regulators {
-			sw1a_reg: sw1a {
-				regulator-min-microvolt = <700000>;
-				regulator-max-microvolt = <1475000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-			/* use sw1c_reg to align with pfuze100/pfuze200 */
-			sw1c_reg: sw1b {
-				regulator-min-microvolt = <700000>;
-				regulator-max-microvolt = <1475000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-
-			sw2_reg: sw2 {
-				regulator-min-microvolt = <1500000>;
-				regulator-max-microvolt = <1850000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw3a_reg: sw3 {
-				regulator-min-microvolt = <900000>;
-				regulator-max-microvolt = <1650000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
+	display0: display {
+		bits-per-pixel = <16>;
+		bus-width = <24>;
 
-			swbst_reg: swbst {
-				regulator-min-microvolt = <5000000>;
-				regulator-max-microvolt = <5150000>;
+		display-timings {
+			native-mode = <&timing0>;
+			timing0: timing0 {
+				clock-frequency = <33500000>;
+				hactive = <800>;
+				vactive = <480>;
+				hback-porch = <89>;
+				hfront-porch = <164>;
+				vback-porch = <23>;
+				vfront-porch = <10>;
+				hsync-len = <10>;
+				vsync-len = <10>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				de-active = <1>;
+				pixelclk-active = <0>;
 			};
+		};
+	};
+};
 
-			snvs_reg: vsnvs {
-				regulator-min-microvolt = <1000000>;
-				regulator-max-microvolt = <3000000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
+&sdma {
+	status = "okay";
+};
 
-			vref_reg: vrefddr {
-				regulator-boot-on;
-				regulator-always-on;
-			};
+&pcie {
+	pinctrl-names = "default";
+	status = "disabled";
+};
 
-			vgen1_reg: vldo1 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
+&pwm1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm1>;
+	status = "okay";
+};
 
-			vgen2_reg: vldo2 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <1550000>;
-			};
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1_1>;
+	status = "okay";
+};
 
-			vgen3_reg: vccsd {
-				regulator-min-microvolt = <2850000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
+&usbh {
+	status = "okay";
+};
 
-			vgen4_reg: v33 {
-				regulator-min-microvolt = <2850000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
+&usbotg1 {
+	vbus-supply = <&reg_usb_otg1_vbus>;
+	srp-disable;
+	hnp-disable;
+	adp-disable;
+	status = "okay";
+};
 
-			vgen5_reg: vldo3 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
+&usbotg2 {
+	vbus-supply = <&reg_usb_otg2_vbus>;
+	srp-disable;
+	hnp-disable;
+	adp-disable;
+	status = "okay";
+};
 
-			vgen6_reg: vldo4 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-		};
-	};
+&usdhc2 {
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc2_1>;
+	pinctrl-1 = <&pinctrl_usdhc2_1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc2_1_200mhz>;
+	assigned-clocks = <&clks IMX7D_USDHC2_ROOT_CLK>;
+	assigned-clocks-rates = <400000000>;
+	bus-width = <8>;
+	tuning-step = <2>;
+	non-removable;
+	keep-power-in-suspend;
+	status = "okay";
 };
 
-&i2c3 {
-	clock-frequency = <100000>;
+&usdhc3 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c3_1>;
+	pinctrl-0 = <&pinctrl_usdhc3_1>;
+	vmmc-supply = <&reg_sd3_vmmc>;
+	cd-gpios = <&gpio1 14>;
+	wp-gpios = <&gpio1 15>;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	no-1-8-v;
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-ecspi.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-ecspi.dts
index de1c2a5..b74863f 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-ecspi.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-ecspi.dts
@@ -8,6 +8,14 @@
 
 #include "imx7d-12x12-lpddr3-arm2.dts"
 
+&epdc {
+        status = "disabled";
+};
+
+&ecspi1{
+        status = "okay";
+};
+
 /*
  * pin conflict with ecspi1
  * default hog setting conflicts with ECSPI1 MOSI and MISO
@@ -16,11 +24,3 @@
 &iomuxc {
 	pinctrl-0 = <&pinctrl_hog_1>;
 };
-
-&epdc {
-        status = "disabled";
-};
-
-&ecspi1{
-        status = "okay";
-};
diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-flexcan.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-flexcan.dts
index 469c7d5..9bc1413 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-flexcan.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-flexcan.dts
@@ -8,10 +8,6 @@
 
 #include "imx7d-12x12-lpddr3-arm2.dts"
 
-&sai1 {
-	status = "disabled";
-};
-
 &fec1 {
 	status = "disabled";
 };
@@ -23,3 +19,7 @@
 &flexcan2 {
 	status = "okay";
 };
+
+&sai1 {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts
index d4d5d25..3ff774f 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts
@@ -11,6 +11,17 @@
 	status = "disabled";
 };
 
+&iomuxc {
+	imx7d-12x12-lpddr3-arm2 {
+		pinctrl_uart2_1: uart2grp-1 {
+			fsl,pins = <
+				MX7D_PAD_UART2_RX_DATA__UART2_DCE_RX 	0x79
+				MX7D_PAD_UART2_TX_DATA__UART2_DCE_TX 	0x79
+			>;
+		};
+	};
+};
+
 &mcctest{
 	status = "okay";
 };
@@ -23,17 +34,6 @@
 	status = "okay";
 };
 
-&iomuxc {
-	uart2 {
-		pinctrl_uart2_1: uart2grp-1 {
-			fsl,pins = <
-				MX7D_PAD_UART2_RX_DATA__UART2_DCE_RX 	0x79
-				MX7D_PAD_UART2_TX_DATA__UART2_DCE_TX 	0x79
-			>;
-		};
-	};
-};
-
 &uart2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart2_1>;
diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-mqs.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-mqs.dts
index 09e75a6..e91ee1e 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-mqs.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-mqs.dts
@@ -33,6 +33,10 @@
 	status = "okay";
 };
 
+&sdma {
+	status = "okay";
+};
+
 &mqs {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_mqs>;
@@ -40,7 +44,3 @@
 	clock-names = "mclk";
 	status = "okay";
 };
-
-&sdma {
-	status = "okay";
-};
diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-sai.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-sai.dts
index 0503cc2..7d989c2 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-sai.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-sai.dts
@@ -25,10 +25,10 @@
 	pinctrl-0 = <&pinctrl_hog_1 &pinctrl_hog>;
 };
 
-&usdhc2 {
-	no-1-8-v;
-};
-
 &sai1 {
 	status = "okay";
 };
+
+&usdhc2 {
+	no-1-8-v;
+};
diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
index d0f7480..b83ce5e 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
@@ -194,6 +194,10 @@
         status = "okay";
 };
 
+&epxp {
+	status = "okay";
+};
+
 &fec1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_enet1>;
@@ -250,11 +254,216 @@
 	status = "disabled";
 };
 
+&i2c1 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c1_1>;
+	status = "okay";
+
+	pmic: pfuze3000@08 {
+		compatible = "fsl,pfuze3000";
+		reg = <0x08>;
+
+		regulators {
+			sw1a_reg: sw1a {
+				regulator-min-microvolt = <700000>;
+				regulator-max-microvolt = <1475000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+			/* use sw1c_reg to align with pfuze100/pfuze200 */
+			sw1c_reg: sw1b {
+				regulator-min-microvolt = <700000>;
+				regulator-max-microvolt = <1475000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw2_reg: sw2 {
+				regulator-min-microvolt = <1500000>;
+				regulator-max-microvolt = <1850000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw3a_reg: sw3 {
+				regulator-min-microvolt = <900000>;
+				regulator-max-microvolt = <1650000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			swbst_reg: swbst {
+				regulator-min-microvolt = <5000000>;
+				regulator-max-microvolt = <5150000>;
+			};
+
+			snvs_reg: vsnvs {
+				regulator-min-microvolt = <1000000>;
+				regulator-max-microvolt = <3000000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vref_reg: vrefddr {
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vgen1_reg: vldo1 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen2_reg: vldo2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+			};
+
+			vgen3_reg: vccsd {
+				regulator-min-microvolt = <2850000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen4_reg: v33 {
+				regulator-min-microvolt = <2850000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen5_reg: vldo3 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen6_reg: vldo4 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+		};
+	};
+};
+
+&i2c3 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c3_1>;
+	status = "okay";
+
+	tsc2007@49 {
+		compatible = "ti,tsc2007";
+		reg = <0x49>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_tsc2007>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <8 0>;
+		pendown-gpio = <&gpio1 8 1>;
+		ti,x-plate-ohms = <660>;
+	};
+
+	max7322: gpio@68 {
+		compatible = "maxim,max7322";
+		reg = <0x68>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+
+	max17135@48 {
+		compatible = "maxim,max17135";
+		reg = <0x48>;
+		vneg_pwrup = <1>;
+		gvee_pwrup = <2>;
+		vpos_pwrup = <10>;
+		gvdd_pwrup = <12>;
+		gvdd_pwrdn = <1>;
+		vpos_pwrdn = <2>;
+		gvee_pwrdn = <8>;
+		vneg_pwrdn = <10>;
+		gpio_pmic_pwrgood = <&gpio2 31 0>;
+		gpio_pmic_vcom_ctrl = <&gpio4 14 0>;
+		gpio_pmic_wakeup = <&gpio4 23 0>;
+		gpio_pmic_v3p3 = <&gpio4 20 0>;
+		gpio_pmic_intr = <&gpio4 18 0>;
+
+		regulators {
+			DISPLAY_reg: DISPLAY {
+				regulator-name = "DISPLAY";
+			};
+
+			GVDD_reg: GVDD {
+				/* 20v */
+				regulator-name = "GVDD";
+			};
+
+			GVEE_reg: GVEE {
+				/* -22v */
+				regulator-name = "GVEE";
+			};
+
+			HVINN_reg: HVINN {
+				/* -22v */
+				regulator-name = "HVINN";
+			};
+
+			HVINP_reg: HVINP {
+				/* 20v */
+				regulator-name = "HVINP";
+			};
+
+			VCOM_reg: VCOM {
+				regulator-name = "VCOM";
+				/* 2's-compliment, -4325000 */
+				regulator-min-microvolt = <0xffbe0178>;
+				/* 2's-compliment, -500000 */
+				regulator-max-microvolt = <0xfff85ee0>;
+			};
+
+			VNEG_reg: VNEG {
+				/* -15v */
+				regulator-name = "VNEG";
+			};
+
+			VPOS_reg: VPOS {
+				/* 15v */
+				regulator-name = "VPOS";
+			};
+
+			V3P3_reg: V3P3 {
+				regulator-name = "V3P3";
+			};
+		};
+	};
+
+	codec: wm8958@1a {
+		compatible = "wlf,wm8958";
+		reg = <0x1a>;
+		clocks = <&clks IMX7D_AUDIO_MCLK_ROOT_CLK>,
+		         <&clks IMX7D_CLK_DUMMY>;
+		clock-names = "mclk1", "mclk2";
+
+		DBVDD1-supply = <&reg_aud_1v8>;
+		DBVDD2-supply = <&reg_aud_1v8>;
+		DBVDD3-supply = <&reg_aud_1v8>;
+		AVDD2-supply = <&reg_aud_1v8>;
+		CPVDD-supply = <&reg_aud_1v8>;
+		SPKVDD1-supply = <&reg_coedc_5v>;
+		SPKVDD2-supply = <&reg_coedc_5v>;
+		wlf,ldo1ena;
+		wlf,ldo2ena;
+	};
+};
+
 &iomuxc {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_hog_1 &pinctrl_hog &pinctrl_hog_sd2_vselect &pinctrl_hog_mipi>;
 
-	hog {
+	imx7d-12x12-lpddr3-arm2 {
 		pinctrl_hog_1: hoggrp-1 {
 			fsl,pins = <
 				MX7D_PAD_EPDC_PWR_STAT__GPIO2_IO31 0x80000000 /* pwrgood */
@@ -292,229 +501,24 @@
 				MX7D_PAD_GPIO1_IO12__SD2_VSELECT  0x59
 			>;
 		};
-	};
 
-	lcdif {
-		pinctrl_lcdif_dat: lcdifdatgrp {
+		pinctrl_ecspi1_cs_1: ecspi1_cs_grp-1 {
 			fsl,pins = <
-				MX7D_PAD_LCD_DATA00__LCD_DATA0	0x79
-				MX7D_PAD_LCD_DATA01__LCD_DATA1	0x79
-				MX7D_PAD_LCD_DATA02__LCD_DATA2	0x79
-				MX7D_PAD_LCD_DATA03__LCD_DATA3	0x79
-				MX7D_PAD_LCD_DATA04__LCD_DATA4	0x79
-				MX7D_PAD_LCD_DATA05__LCD_DATA5	0x79
-				MX7D_PAD_LCD_DATA06__LCD_DATA6	0x79
-				MX7D_PAD_LCD_DATA07__LCD_DATA7	0x79
-				MX7D_PAD_LCD_DATA08__LCD_DATA8	0x79
-				MX7D_PAD_LCD_DATA09__LCD_DATA9	0x79
-				MX7D_PAD_LCD_DATA10__LCD_DATA10	0x79
-				MX7D_PAD_LCD_DATA11__LCD_DATA11	0x79
-				MX7D_PAD_LCD_DATA12__LCD_DATA12	0x79
-				MX7D_PAD_LCD_DATA13__LCD_DATA13	0x79
-				MX7D_PAD_LCD_DATA14__LCD_DATA14	0x79
-				MX7D_PAD_LCD_DATA15__LCD_DATA15	0x79
-				MX7D_PAD_LCD_DATA16__LCD_DATA16	0x79
-				MX7D_PAD_LCD_DATA17__LCD_DATA17	0x79
-				MX7D_PAD_LCD_DATA18__LCD_DATA18	0x79
-				MX7D_PAD_LCD_DATA19__LCD_DATA19	0x79
-				MX7D_PAD_LCD_DATA20__LCD_DATA20	0x79
-				MX7D_PAD_LCD_DATA21__LCD_DATA21	0x79
-				MX7D_PAD_LCD_DATA22__LCD_DATA22	0x79
-				MX7D_PAD_LCD_DATA23__LCD_DATA23	0x79
+				MX7D_PAD_ECSPI1_SS0__GPIO4_IO19     0x2
 			>;
 		};
 
-		pinctrl_lcdif_ctrl: lcdifctrlgrp {
+		pinctrl_ecspi1_1: ecspi1grp-1 {
 			fsl,pins = <
-				MX7D_PAD_LCD_CLK__LCD_CLK	0x79
-				MX7D_PAD_LCD_ENABLE__LCD_ENABLE	0x79
-				MX7D_PAD_LCD_VSYNC__LCD_VSYNC	0x79
-				MX7D_PAD_LCD_HSYNC__LCD_HSYNC	0x79
+				MX7D_PAD_ECSPI1_MISO__ECSPI1_MISO   0x2
+				MX7D_PAD_ECSPI1_MOSI__ECSPI1_MOSI   0x2
+				MX7D_PAD_ECSPI1_SCLK__ECSPI1_SCLK   0x2
 			>;
 		};
-	};
 
-	pcie {
-		pinctrl_pcie: pciegrp {
+		pinctrl_ecspi3_cs_1: ecspi3_cs_grp-1 {
 			fsl,pins = <
-				MX7D_PAD_SAI2_RX_DATA__GPIO6_IO21	0x2
-			>;
-		};
-
-	};
-};
-
-&lcdif {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lcdif_dat
-		     &pinctrl_lcdif_ctrl>;
-	display = <&display0>;
-	status = "okay";
-
-	display0: display {
-		bits-per-pixel = <16>;
-		bus-width = <24>;
-
-		display-timings {
-			native-mode = <&timing0>;
-			timing0: timing0 {
-				clock-frequency = <33500000>;
-				hactive = <800>;
-				vactive = <480>;
-				hback-porch = <89>;
-				hfront-porch = <164>;
-				vback-porch = <23>;
-				vfront-porch = <10>;
-				hsync-len = <10>;
-				vsync-len = <10>;
-				hsync-active = <0>;
-				vsync-active = <0>;
-				de-active = <1>;
-				pixelclk-active = <0>;
-			};
-		};
-	};
-};
-
-&pcie {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pcie>;
-	reset-gpio = <&gpio6 21 0>;
-	status = "disabled";
-};
-
-&pwm1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pwm1>;
-	status = "okay";
-};
-
-&sim1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_sim1_1>;
-	status = "disabled";
-};
-
-&uart1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart1_1>;
-	assigned-clocks = <&clks IMX7D_UART1_ROOT_SRC>;
-	assigned-clock-parents = <&clks IMX7D_OSC_24M_CLK>;
-	status = "okay";
-};
-
-&uart3 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart3_1>;
-	fsl,uart-has-rtscts;
-	assigned-clocks = <&clks IMX7D_UART3_ROOT_SRC>;
-	assigned-clock-parents = <&clks IMX7D_PLL_SYS_MAIN_240M_CLK>;
-	status = "okay";
-	/* for DTE mode, add below change */
-	/* fsl,dte-mode;*/
-	/* pinctrl-0 = <&pinctrl_uart3dte_1>; */
-};
-
-&iomuxc_lpsr {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hog_2>;
-
-	hog {
-		pinctrl_hog_2: hoggrp-2 {
-			fsl,pins = <
-				MX7D_PAD_GPIO1_IO07__GPIO1_IO7	0x14
-				MX7D_PAD_GPIO1_IO05__GPIO1_IO5	0x14
-				MX7D_PAD_GPIO1_IO00__WDOD1_WDOG_B 0x74
-			>;
-		};
-	};
-
-	pwm1 {
-		pinctrl_pwm1: pwm1grp {
-			fsl,pins = <
-				MX7D_PAD_GPIO1_IO01__PWM1_OUT	0x110b0
-			>;
-		};
-	};
-};
-
-&usbotg1 {
-	vbus-supply = <&reg_usb_otg1_vbus>;
-	srp-disable;
-	hnp-disable;
-	adp-disable;
-	status = "okay";
-};
-
-&usbotg2 {
-	vbus-supply = <&reg_usb_otg2_vbus>;
-	srp-disable;
-	hnp-disable;
-	adp-disable;
-	status = "okay";
-};
-
-&usdhc1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_usdhc1_1>;
-	cd-gpios = <&gpio5 0 0>;
-	wp-gpios = <&gpio5 1 0>;
-	no-1-8-v;
-	keep-power-in-suspend;
-	enable-sdio-wakeup;
-	vmmc-supply = <&reg_sd1_vmmc>;
-	status = "okay";
-};
-
-&usdhc2 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
-	pinctrl-0 = <&pinctrl_usdhc2_1>;
-	pinctrl-1 = <&pinctrl_usdhc2_1_100mhz>;
-	pinctrl-2 = <&pinctrl_usdhc2_1_200mhz>;
-	cd-gpios = <&gpio5 9 0>;
-	wp-gpios = <&gpio5 10 0>;
-	keep-power-in-suspend;
-	enable-sdio-wakeup;
-	tuning-step = <2>;
-	vmmc-supply = <&reg_sd2_vmmc>;
-	status = "okay";
-};
-
-&usdhc3 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
-	pinctrl-0 = <&pinctrl_usdhc3_1>;
-	pinctrl-1 = <&pinctrl_usdhc3_1_100mhz>;
-	pinctrl-2 = <&pinctrl_usdhc3_1_200mhz>;
-	assigned-clocks = <&clks IMX7D_USDHC3_ROOT_CLK>;
-	assigned-clock-rates = <400000000>;
-	bus-width = <8>;
-	tuning-step = <2>;
-	non-removable;
-	keep-power-in-suspend;
-	status = "okay";
-};
-
-&iomuxc {
-	ecspi1 {
-		pinctrl_ecspi1_cs_1: ecspi1_cs_grp-1 {
-			fsl,pins = <
-				MX7D_PAD_ECSPI1_SS0__GPIO4_IO19     0x2
-			>;
-		};
-
-		pinctrl_ecspi1_1: ecspi1grp-1 {
-			fsl,pins = <
-				MX7D_PAD_ECSPI1_MISO__ECSPI1_MISO   0x2
-				MX7D_PAD_ECSPI1_MOSI__ECSPI1_MOSI   0x2
-				MX7D_PAD_ECSPI1_SCLK__ECSPI1_SCLK   0x2
-			>;
-		};
-	};
-
-	ecspi3 {
-		pinctrl_ecspi3_cs_1: ecspi3_cs_grp-1 {
-			fsl,pins = <
-				MX7D_PAD_SAI2_TX_DATA__GPIO6_IO22	0x2
+				MX7D_PAD_SAI2_TX_DATA__GPIO6_IO22	0x2
 			>;
 		};
 
@@ -525,9 +529,7 @@
 				MX7D_PAD_SAI2_RX_DATA__ECSPI3_SCLK	0x2
 			>;
 		};
-	};
 
-	enet1 {
 		pinctrl_enet1: enet1grp {
 			fsl,pins = <
 				MX7D_PAD_GPIO1_IO10__ENET1_MDIO			0x3
@@ -547,9 +549,6 @@
 			>;
 		};
 
-	};
-
-	enet2 {
 		pinctrl_enet2: enet2grp {
 			fsl,pins = <
 				MX7D_PAD_EPDC_GDSP__ENET2_RGMII_TXC     0x1
@@ -566,11 +565,9 @@
 				MX7D_PAD_EPDC_SDCE0__ENET2_RGMII_RX_CTL 0x1
 			>;
 		};
-	};
 
-	epdc {
-                pinctrl_epdc_0: epdcgrp-0 {
-                        fsl,pins = <
+		pinctrl_epdc_0: epdcgrp-0 {
+			fsl,pins = <
 				MX7D_PAD_EPDC_DATA00__EPDC_DATA0  0x2
 				MX7D_PAD_EPDC_DATA01__EPDC_DATA1  0x2
 				MX7D_PAD_EPDC_DATA02__EPDC_DATA2  0x2
@@ -601,11 +598,9 @@
 				MX7D_PAD_EPDC_GDSP__EPDC_GDSP     0x2
 				MX7D_PAD_EPDC_BDR0__EPDC_BDR0     0x2
 				MX7D_PAD_EPDC_BDR1__EPDC_BDR1     0x2
-                        >;
-                };
-	};
+			>;
+		};
 
-	flexcan1 {
 		pinctrl_flexcan1: flexcan1grp {
 			fsl,pins = <
 				MX7D_PAD_SAI1_RX_DATA__FLEXCAN1_RX	0x59
@@ -613,9 +608,7 @@
 				MX7D_PAD_GPIO1_IO10__GPIO1_IO10		0x59 /* STBY */
 			>;
 		};
-	};
 
-	flexcan2 {
 		pinctrl_flexcan2: flexcan2grp {
 			fsl,pins = <
 				MX7D_PAD_SAI1_TX_SYNC__FLEXCAN2_RX	0x59
@@ -623,18 +616,14 @@
 				MX7D_PAD_GPIO1_IO11__GPIO1_IO11		0x59  /* STBY */
 			>;
 		};
-	};
 
-	gpio_keys {
 		pinctrl_gpio_keys: gpio_keysgrp {
 			fsl,pins = <
 				MX7D_PAD_GPIO1_IO14__GPIO1_IO14 0x32
 				MX7D_PAD_GPIO1_IO15__GPIO1_IO15 0x32
 			>;
 		};
-	};
 
-	i2c1 {
 		pinctrl_i2c1_1: i2c1grp-1 {
 			fsl,pins = <
 				MX7D_PAD_I2C1_SDA__I2C1_SDA          0x4000007f
@@ -642,18 +631,13 @@
 			>;
 		};
 
-	};
-
-	i2c2 {
 		pinctrl_i2c2_1: i2c2grp-1 {
 			fsl,pins = <
 				MX7D_PAD_I2C2_SDA__I2C2_SDA          0x4000007f
 				MX7D_PAD_I2C2_SCL__I2C2_SCL          0x4000007f
 			>;
 		};
-	};
 
-	i2c3 {
 		pinctrl_i2c3_1: i2c3grp-1 {
 			fsl,pins = <
 				MX7D_PAD_I2C3_SDA__I2C3_SDA	     0x4000007f
@@ -661,27 +645,86 @@
 			>;
 		};
 
-	};
-
-	i2c4 {
 		pinctrl_i2c4_1: i2c4grp-1 {
 			fsl,pins = <
 				MX7D_PAD_I2C4_SDA__I2C4_SDA          0x4000007f
 				MX7D_PAD_I2C4_SCL__I2C4_SCL          0x4000007f
 			>;
 		};
-	};
 
-	mqs {
+		pinctrl_lcdif_dat: lcdifdatgrp {
+			fsl,pins = <
+				MX7D_PAD_LCD_DATA00__LCD_DATA0	0x79
+				MX7D_PAD_LCD_DATA01__LCD_DATA1	0x79
+				MX7D_PAD_LCD_DATA02__LCD_DATA2	0x79
+				MX7D_PAD_LCD_DATA03__LCD_DATA3	0x79
+				MX7D_PAD_LCD_DATA04__LCD_DATA4	0x79
+				MX7D_PAD_LCD_DATA05__LCD_DATA5	0x79
+				MX7D_PAD_LCD_DATA06__LCD_DATA6	0x79
+				MX7D_PAD_LCD_DATA07__LCD_DATA7	0x79
+				MX7D_PAD_LCD_DATA08__LCD_DATA8	0x79
+				MX7D_PAD_LCD_DATA09__LCD_DATA9	0x79
+				MX7D_PAD_LCD_DATA10__LCD_DATA10	0x79
+				MX7D_PAD_LCD_DATA11__LCD_DATA11	0x79
+				MX7D_PAD_LCD_DATA12__LCD_DATA12	0x79
+				MX7D_PAD_LCD_DATA13__LCD_DATA13	0x79
+				MX7D_PAD_LCD_DATA14__LCD_DATA14	0x79
+				MX7D_PAD_LCD_DATA15__LCD_DATA15	0x79
+				MX7D_PAD_LCD_DATA16__LCD_DATA16	0x79
+				MX7D_PAD_LCD_DATA17__LCD_DATA17	0x79
+				MX7D_PAD_LCD_DATA18__LCD_DATA18	0x79
+				MX7D_PAD_LCD_DATA19__LCD_DATA19	0x79
+				MX7D_PAD_LCD_DATA20__LCD_DATA20	0x79
+				MX7D_PAD_LCD_DATA21__LCD_DATA21	0x79
+				MX7D_PAD_LCD_DATA22__LCD_DATA22	0x79
+				MX7D_PAD_LCD_DATA23__LCD_DATA23	0x79
+			>;
+		};
+
+		pinctrl_lcdif_ctrl: lcdifctrlgrp {
+			fsl,pins = <
+				MX7D_PAD_LCD_CLK__LCD_CLK	0x79
+				MX7D_PAD_LCD_ENABLE__LCD_ENABLE	0x79
+				MX7D_PAD_LCD_VSYNC__LCD_VSYNC	0x79
+				MX7D_PAD_LCD_HSYNC__LCD_HSYNC	0x79
+			>;
+		};
+
 		pinctrl_mqs: mqsgrp {
 			fsl,pins = <
 				MX7D_PAD_SAI1_RX_SYNC__MQS_RIGHT     0x0
 				MX7D_PAD_SAI1_RX_BCLK__MQS_LEFT      0x0
 			>;
 		};
-	};
 
-	sim1 {
+		pinctrl_pcie: pciegrp {
+			fsl,pins = <
+				MX7D_PAD_SAI2_RX_DATA__GPIO6_IO21	0x2
+			>;
+		};
+
+		pinctrl_sai1: sai1grp {
+			fsl,pins = <
+				MX7D_PAD_SAI1_MCLK__SAI1_MCLK           0x1f
+				MX7D_PAD_SAI1_TX_BCLK__SAI1_TX_BCLK     0x1f
+				MX7D_PAD_SAI1_TX_SYNC__SAI1_TX_SYNC     0x1f
+				MX7D_PAD_SAI1_RX_SYNC__SAI1_RX_SYNC     0x1f
+				MX7D_PAD_SAI1_RX_DATA__SAI1_RX_DATA0    0x1f
+				MX7D_PAD_SAI1_TX_DATA__SAI1_TX_DATA0    0
+				MX7D_PAD_GPIO1_IO12__GPIO1_IO12		0x59
+				MX7D_PAD_GPIO1_IO13__GPIO1_IO13		0x59
+			>;
+		};
+
+		pinctrl_sai2: sai2grp {
+			fsl,pins = <
+				MX7D_PAD_SAI2_TX_BCLK__SAI2_TX_BCLK     0x1f
+				MX7D_PAD_SAI2_TX_SYNC__SAI2_TX_SYNC     0x1f
+				MX7D_PAD_SAI2_RX_DATA__SAI2_RX_DATA0    0x1f
+				MX7D_PAD_SAI2_TX_DATA__SAI2_TX_DATA0    0
+			>;
+		};
+
 		pinctrl_sim1_1: sim1grp-1 {
 			fsl,pins = <
 				MX7D_PAD_SAI1_TX_SYNC__SIM1_PORT1_RST_B		0x77
@@ -692,17 +735,13 @@
 
 			>;
 		};
-	};
 
-	touch {
 		pinctrl_tsc2007: tsc2007grp {
 			fsl,pins = <
 				MX7D_PAD_GPIO1_IO08__GPIO1_IO8	0x80000000
 			>;
 		};
-	};
 
-	uart1 {
 		pinctrl_uart1_1: uart1grp-1 {
 			fsl,pins = <
 				MX7D_PAD_UART1_TX_DATA__UART1_DCE_TX 0x79
@@ -710,9 +749,6 @@
 			>;
 		};
 
-	};
-
-	uart3 {
 		pinctrl_uart3_1: uart3grp-1 {
 			fsl,pins = <
 				MX7D_PAD_UART3_TX_DATA__UART3_DCE_TX    0x79
@@ -730,9 +766,7 @@
 				MX7D_PAD_UART3_CTS_B__UART3_DTE_RTS     0x79
 			>;
 		};
-	};
 
-	usdhc1 {
 		pinctrl_usdhc1_1: usdhc1grp-1 {
 			fsl,pins = <
 				MX7D_PAD_SD1_CMD__SD1_CMD	0x59
@@ -743,9 +777,7 @@
 				MX7D_PAD_SD1_DATA3__SD1_DATA3	0x59
 			>;
 		};
-	};
 
-	usdhc2 {
 		pinctrl_usdhc2_1: usdhc2grp-1 {
 			fsl,pins = <
 				MX7D_PAD_SD2_CMD__SD2_CMD     0x59
@@ -778,9 +810,7 @@
 				MX7D_PAD_SD2_DATA3__SD2_DATA3 0x5b
 			>;
 		};
-	};
 
-	usdhc3 {
 		pinctrl_usdhc3_1: usdhc3grp-1 {
 			fsl,pins = <
 				MX7D_PAD_SD3_CMD__SD3_CMD	    0x59
@@ -829,34 +859,74 @@
 			>;
 		};
 	};
+};
 
-	sai1 {
-		pinctrl_sai1: sai1grp {
+&iomuxc_lpsr {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog_2>;
+
+	imx7d-12x12-lpddr3-arm2 {
+		pinctrl_hog_2: hoggrp-2 {
 			fsl,pins = <
-				MX7D_PAD_SAI1_MCLK__SAI1_MCLK           0x1f
-				MX7D_PAD_SAI1_TX_BCLK__SAI1_TX_BCLK     0x1f
-				MX7D_PAD_SAI1_TX_SYNC__SAI1_TX_SYNC     0x1f
-				MX7D_PAD_SAI1_RX_SYNC__SAI1_RX_SYNC     0x1f
-				MX7D_PAD_SAI1_RX_DATA__SAI1_RX_DATA0    0x1f
-				MX7D_PAD_SAI1_TX_DATA__SAI1_TX_DATA0    0
-				MX7D_PAD_GPIO1_IO12__GPIO1_IO12		0x59
-				MX7D_PAD_GPIO1_IO13__GPIO1_IO13		0x59
+				MX7D_PAD_GPIO1_IO07__GPIO1_IO7	0x14
+				MX7D_PAD_GPIO1_IO05__GPIO1_IO5	0x14
+				MX7D_PAD_GPIO1_IO00__WDOD1_WDOG_B 0x74
 			>;
 		};
-	};
 
-	sai2 {
-		pinctrl_sai2: sai2grp {
+		pinctrl_pwm1: pwm1grp {
 			fsl,pins = <
-				MX7D_PAD_SAI2_TX_BCLK__SAI2_TX_BCLK     0x1f
-				MX7D_PAD_SAI2_TX_SYNC__SAI2_TX_SYNC     0x1f
-				MX7D_PAD_SAI2_RX_DATA__SAI2_RX_DATA0    0x1f
-				MX7D_PAD_SAI2_TX_DATA__SAI2_TX_DATA0    0
+				MX7D_PAD_GPIO1_IO01__PWM1_OUT	0x110b0
 			>;
 		};
 	};
 };
 
+&lcdif {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lcdif_dat
+		     &pinctrl_lcdif_ctrl>;
+	display = <&display0>;
+	status = "okay";
+
+	display0: display {
+		bits-per-pixel = <16>;
+		bus-width = <24>;
+
+		display-timings {
+			native-mode = <&timing0>;
+			timing0: timing0 {
+				clock-frequency = <33500000>;
+				hactive = <800>;
+				vactive = <480>;
+				hback-porch = <89>;
+				hfront-porch = <164>;
+				vback-porch = <23>;
+				vfront-porch = <10>;
+				hsync-len = <10>;
+				vsync-len = <10>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				de-active = <1>;
+				pixelclk-active = <0>;
+			};
+		};
+	};
+};
+
+&pcie {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pcie>;
+	reset-gpio = <&gpio6 21 0>;
+	status = "disabled";
+};
+
+&pwm1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm1>;
+	status = "okay";
+};
+
 &sai1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_sai1>;
@@ -873,211 +943,84 @@
 	status = "okay";
 };
 
-&epxp {
-	status = "okay";
+&sim1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_sim1_1>;
+	status = "disabled";
 };
 
-&i2c1 {
-	clock-frequency = <100000>;
+&uart1 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c1_1>;
+	pinctrl-0 = <&pinctrl_uart1_1>;
+	assigned-clocks = <&clks IMX7D_UART1_ROOT_SRC>;
+	assigned-clock-parents = <&clks IMX7D_OSC_24M_CLK>;
 	status = "okay";
-
-	pmic: pfuze3000@08 {
-		compatible = "fsl,pfuze3000";
-		reg = <0x08>;
-
-		regulators {
-			sw1a_reg: sw1a {
-				regulator-min-microvolt = <700000>;
-				regulator-max-microvolt = <1475000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-			/* use sw1c_reg to align with pfuze100/pfuze200 */
-			sw1c_reg: sw1b {
-				regulator-min-microvolt = <700000>;
-				regulator-max-microvolt = <1475000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-
-			sw2_reg: sw2 {
-				regulator-min-microvolt = <1500000>;
-				regulator-max-microvolt = <1850000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw3a_reg: sw3 {
-				regulator-min-microvolt = <900000>;
-				regulator-max-microvolt = <1650000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			swbst_reg: swbst {
-				regulator-min-microvolt = <5000000>;
-				regulator-max-microvolt = <5150000>;
-			};
-
-			snvs_reg: vsnvs {
-				regulator-min-microvolt = <1000000>;
-				regulator-max-microvolt = <3000000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vref_reg: vrefddr {
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vgen1_reg: vldo1 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen2_reg: vldo2 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <1550000>;
-			};
-
-			vgen3_reg: vccsd {
-				regulator-min-microvolt = <2850000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen4_reg: v33 {
-				regulator-min-microvolt = <2850000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen5_reg: vldo3 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen6_reg: vldo4 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-		};
-	};
 };
 
-&i2c3 {
-	clock-frequency = <100000>;
+&uart3 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c3_1>;
+	pinctrl-0 = <&pinctrl_uart3_1>;
+	fsl,uart-has-rtscts;
+	assigned-clocks = <&clks IMX7D_UART3_ROOT_SRC>;
+	assigned-clock-parents = <&clks IMX7D_PLL_SYS_MAIN_240M_CLK>;
 	status = "okay";
+	/* for DTE mode, add below change */
+	/* fsl,dte-mode;*/
+	/* pinctrl-0 = <&pinctrl_uart3dte_1>; */
+};
 
-	tsc2007@49 {
-		compatible = "ti,tsc2007";
-		reg = <0x49>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_tsc2007>;
-		interrupt-parent = <&gpio1>;
-		interrupts = <8 0>;
-		pendown-gpio = <&gpio1 8 1>;
-		ti,x-plate-ohms = <660>;
-	};
-
-	max7322: gpio@68 {
-		compatible = "maxim,max7322";
-		reg = <0x68>;
-		gpio-controller;
-		#gpio-cells = <2>;
-	};
-
-	max17135@48 {
-		compatible = "maxim,max17135";
-		reg = <0x48>;
-		vneg_pwrup = <1>;
-		gvee_pwrup = <2>;
-		vpos_pwrup = <10>;
-		gvdd_pwrup = <12>;
-		gvdd_pwrdn = <1>;
-		vpos_pwrdn = <2>;
-		gvee_pwrdn = <8>;
-		vneg_pwrdn = <10>;
-		gpio_pmic_pwrgood = <&gpio2 31 0>;
-		gpio_pmic_vcom_ctrl = <&gpio4 14 0>;
-		gpio_pmic_wakeup = <&gpio4 23 0>;
-		gpio_pmic_v3p3 = <&gpio4 20 0>;
-		gpio_pmic_intr = <&gpio4 18 0>;
-
-		regulators {
-			DISPLAY_reg: DISPLAY {
-				regulator-name = "DISPLAY";
-			};
-
-			GVDD_reg: GVDD {
-				/* 20v */
-				regulator-name = "GVDD";
-			};
-
-			GVEE_reg: GVEE {
-				/* -22v */
-				regulator-name = "GVEE";
-			};
-
-			HVINN_reg: HVINN {
-				/* -22v */
-				regulator-name = "HVINN";
-			};
-
-			HVINP_reg: HVINP {
-				/* 20v */
-				regulator-name = "HVINP";
-			};
-
-			VCOM_reg: VCOM {
-				regulator-name = "VCOM";
-				/* 2's-compliment, -4325000 */
-				regulator-min-microvolt = <0xffbe0178>;
-				/* 2's-compliment, -500000 */
-				regulator-max-microvolt = <0xfff85ee0>;
-			};
-
-			VNEG_reg: VNEG {
-				/* -15v */
-				regulator-name = "VNEG";
-			};
+&usbotg1 {
+	vbus-supply = <&reg_usb_otg1_vbus>;
+	srp-disable;
+	hnp-disable;
+	adp-disable;
+	status = "okay";
+};
 
-			VPOS_reg: VPOS {
-				/* 15v */
-				regulator-name = "VPOS";
-			};
+&usbotg2 {
+	vbus-supply = <&reg_usb_otg2_vbus>;
+	srp-disable;
+	hnp-disable;
+	adp-disable;
+	status = "okay";
+};
 
-			V3P3_reg: V3P3 {
-				regulator-name = "V3P3";
-			};
-		};
-	};
+&usdhc1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc1_1>;
+	cd-gpios = <&gpio5 0 0>;
+	wp-gpios = <&gpio5 1 0>;
+	no-1-8-v;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	vmmc-supply = <&reg_sd1_vmmc>;
+	status = "okay";
+};
 
-	codec: wm8958@1a {
-		compatible = "wlf,wm8958";
-		reg = <0x1a>;
-		clocks = <&clks IMX7D_AUDIO_MCLK_ROOT_CLK>,
-		         <&clks IMX7D_CLK_DUMMY>;
-		clock-names = "mclk1", "mclk2";
+&usdhc2 {
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc2_1>;
+	pinctrl-1 = <&pinctrl_usdhc2_1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc2_1_200mhz>;
+	cd-gpios = <&gpio5 9 0>;
+	wp-gpios = <&gpio5 10 0>;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	tuning-step = <2>;
+	vmmc-supply = <&reg_sd2_vmmc>;
+	status = "okay";
+};
 
-		DBVDD1-supply = <&reg_aud_1v8>;
-		DBVDD2-supply = <&reg_aud_1v8>;
-		DBVDD3-supply = <&reg_aud_1v8>;
-		AVDD2-supply = <&reg_aud_1v8>;
-		CPVDD-supply = <&reg_aud_1v8>;
-		SPKVDD1-supply = <&reg_coedc_5v>;
-		SPKVDD2-supply = <&reg_coedc_5v>;
-		wlf,ldo1ena;
-		wlf,ldo2ena;
-	};
+&usdhc3 {
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc3_1>;
+	pinctrl-1 = <&pinctrl_usdhc3_1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc3_1_200mhz>;
+	assigned-clocks = <&clks IMX7D_USDHC3_ROOT_CLK>;
+	assigned-clock-rates = <400000000>;
+	bus-width = <8>;
+	tuning-step = <2>;
+	non-removable;
+	keep-power-in-suspend;
+	status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-csi.dts b/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-csi.dts
index facec64..69d0d98 100644
--- a/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-csi.dts
+++ b/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-csi.dts
@@ -8,10 +8,6 @@
 
 #include "imx7d-19x19-ddr3-arm2.dts"
 
-&lcdif {
-	status = "disabled";
-};
-
 &csi1 {
 	status = "okay";
 
@@ -44,3 +40,7 @@
 	};
 };
 
+&lcdif {
+	status = "disabled";
+};
+
diff --git a/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-mipi.dts b/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-mipi.dts
index 33cbf54..6f3a2096 100644
--- a/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-mipi.dts
+++ b/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-mipi.dts
@@ -19,23 +19,6 @@
 	};
 };
 
-&mipi_csi {
-	clock-frequency = <240000000>;
-	status = "okay";
-	port {
-		mipi_sensor_ep: endpoint1 {
-			remote-endpoint = <&ov5640_mipi_ep>;
-			data-lanes = <2>;
-			csis-hs-settle = <13>;
-			csis-wclk;
-		};
-
-		csi_mipi_ep: endpoint2 {
-			remote-endpoint = <&csi_ep>;
-		};
-	};
-};
-
 &i2c2 {
 	ov5640_mipi: ov5640_mipi@3c {
 		compatible = "ovti,ov5640_mipi";
@@ -57,3 +40,20 @@
 	};
 };
 
+&mipi_csi {
+	clock-frequency = <240000000>;
+	status = "okay";
+	port {
+		mipi_sensor_ep: endpoint1 {
+			remote-endpoint = <&ov5640_mipi_ep>;
+			data-lanes = <2>;
+			csis-hs-settle = <13>;
+			csis-wclk;
+		};
+
+		csi_mipi_ep: endpoint2 {
+			remote-endpoint = <&csi_ep>;
+		};
+	};
+};
+
diff --git a/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-sai.dts b/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-sai.dts
index 7f23331..1887233 100644
--- a/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-sai.dts
+++ b/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2-sai.dts
@@ -24,10 +24,10 @@
 	pinctrl-0 = <&pinctrl_hog_1>;
 };
 
-&usdhc3 {
-	no-1-8-v;
-};
-
 &sai2 {
 	status = "okay";
 };
+
+&usdhc3 {
+	no-1-8-v;
+};
diff --git a/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2.dts b/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2.dts
index 4790a3f..caa4b141 100644
--- a/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-19x19-ddr3-arm2.dts
@@ -151,6 +151,26 @@
 	arm-supply = <&sw1a_reg>;
 };
 
+&ecspi1 {
+	    fsl,spi-num-chipselects = <1>;
+		cs-gpios = <&gpio4 19 0>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_ecspi1_1 &pinctrl_ecspi1_cs_1>;
+		status = "okay";
+
+	    flash: at45@0 {
+	        #address-cells = <1>;
+	        #size-cells = <1>;
+	        compatible = "atmel,at45";
+	        spi-max-frequency = <20000000>;
+	        reg = <0>;
+	    };
+};
+
+&epxp {
+        status = "okay";
+};
+
 &fec1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_enet1>;
@@ -194,115 +214,169 @@
 	status = "okay";
 };
 
-&iomuxc {
+&i2c1 {
+	clock-frequency = <100000>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hog_1 &pinctrl_hog_sd3_vselect>;
+	pinctrl-0 = <&pinctrl_i2c1_1>;
+	status = "okay";
 
-	hog {
-		pinctrl_hog_1: hoggrp-1 {
-			fsl,pins = <
-				MX7D_PAD_GPIO1_IO12__SD2_VSELECT   0x59
-				MX7D_PAD_SD2_CD_B__GPIO5_IO9       0x59
-				MX7D_PAD_SD2_WP__GPIO5_IO10        0x59
-				MX7D_PAD_SD2_RESET_B__GPIO5_IO11   0x59
-				MX7D_PAD_SD3_RESET_B__GPIO6_IO11   0x59
-				MX7D_PAD_GPIO1_IO14__GPIO1_IO14    0x59
-				MX7D_PAD_GPIO1_IO15__GPIO1_IO15    0x59
-				MX7D_PAD_SD3_STROBE__GPIO6_IO10    0x59
+	pmic: pfuze3000@08 {
+		compatible = "fsl,pfuze3000";
+		reg = <0x08>;
 
-				MX7D_PAD_I2C4_SCL__GPIO4_IO14	  0x7F
-				MX7D_PAD_I2C4_SDA__GPIO4_IO15     0x7F
-			>;
-		};
+		regulators {
+			sw1a_reg: sw1a {
+				regulator-min-microvolt = <700000>;
+				regulator-max-microvolt = <1475000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+			/* use sw1c_reg to align with pfuze100/pfuze200 */
+			sw1c_reg: sw1b {
+				regulator-min-microvolt = <700000>;
+				regulator-max-microvolt = <1475000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
 
-		pinctrl_hog_sd3_vselect: hoggrp_sd3vselect {
-			fsl,pins = <
-				MX7D_PAD_GPIO1_IO13__SD3_VSELECT   0x59
-			>;
+			sw2_reg: sw2 {
+				regulator-min-microvolt = <1500000>;
+				regulator-max-microvolt = <1850000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw3a_reg: sw3 {
+				regulator-min-microvolt = <900000>;
+				regulator-max-microvolt = <1650000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			swbst_reg: swbst {
+				regulator-min-microvolt = <5000000>;
+				regulator-max-microvolt = <5150000>;
+			};
+
+			snvs_reg: vsnvs {
+				regulator-min-microvolt = <1000000>;
+				regulator-max-microvolt = <3000000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vref_reg: vrefddr {
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vgen1_reg: vldo1 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen2_reg: vldo2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+			};
+
+			vgen3_reg: vccsd {
+				regulator-min-microvolt = <2850000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen4_reg: v33 {
+				regulator-min-microvolt = <2850000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen5_reg: vldo3 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen6_reg: vldo4 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
 		};
 	};
 };
 
-&uart1 {
+&i2c2 {
+	clock-frequency = <100000>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart1_1>;
-	assigned-clocks = <&clks IMX7D_UART1_ROOT_SRC>;
-	assigned-clock-parents = <&clks IMX7D_OSC_24M_CLK>;
+	pinctrl-0 = <&pinctrl_i2c2_1>;
 	status = "okay";
 };
 
-&uart3 {
+&i2c3 {
+	clock-frequency = <100000>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart3_1>;
-	fsl,uart-has-rtscts;
-	assigned-clocks = <&clks IMX7D_UART3_ROOT_SRC>;
-	assigned-clock-parents = <&clks IMX7D_PLL_SYS_MAIN_240M_CLK>;
+	pinctrl-0 = <&pinctrl_i2c3_1>;
 	status = "okay";
-	/* for DTE mode, add below change */
-	/* fsl,dte-mode;*/
-	pinctrl-0 = <&pinctrl_uart3dte_1>;
+
+	max7322: gpio@68 {
+		compatible = "maxim,max7322";
+		reg = <0x68>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		resets = <&max7322_reset>;
+	};
+	codec: wm8958@1a {
+		compatible = "wlf,wm8958";
+		reg = <0x1a>;
+		clocks = <&clks IMX7D_AUDIO_MCLK_ROOT_CLK>,
+		       <&clks IMX7D_CLK_DUMMY>;
+		clock-names = "mclk1", "mclk2";
+
+		DBVDD1-supply = <&reg_aud_1v8>;
+		DBVDD2-supply = <&reg_aud_1v8>;
+		DBVDD3-supply = <&reg_aud_3v3>;
+		AVDD2-supply = <&reg_aud_1v8>;
+		CPVDD-supply = <&reg_aud_1v8>;
+		SPKVDD1-supply = <&reg_coedc_5v>;
+		SPKVDD2-supply = <&reg_coedc_5v>;
+
+		wlf,ldo1ena;
+		wlf,ldo2ena;
+	};
 };
 
-&iomuxc_lpsr {
+&iomuxc {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hog_2>;
+	pinctrl-0 = <&pinctrl_hog_1 &pinctrl_hog_sd3_vselect>;
 
-	hog {
-		pinctrl_hog_2: hoggrp-2 {
+	imx7d-19x19-ddr3-arm2 {
+		pinctrl_hog_1: hoggrp-1 {
 			fsl,pins = <
-				MX7D_PAD_GPIO1_IO03__GPIO1_IO3 0x80000000
-				MX7D_PAD_GPIO1_IO07__GPIO1_IO7 0x80000000
+				MX7D_PAD_GPIO1_IO12__SD2_VSELECT   0x59
+				MX7D_PAD_SD2_CD_B__GPIO5_IO9       0x59
+				MX7D_PAD_SD2_WP__GPIO5_IO10        0x59
+				MX7D_PAD_SD2_RESET_B__GPIO5_IO11   0x59
+				MX7D_PAD_SD3_RESET_B__GPIO6_IO11   0x59
+				MX7D_PAD_GPIO1_IO14__GPIO1_IO14    0x59
+				MX7D_PAD_GPIO1_IO15__GPIO1_IO15    0x59
+				MX7D_PAD_SD3_STROBE__GPIO6_IO10    0x59
+
+				MX7D_PAD_I2C4_SCL__GPIO4_IO14	  0x7F
+				MX7D_PAD_I2C4_SDA__GPIO4_IO15     0x7F
 			>;
 		};
-	};
 
-	mipi_csi {
-		pinctrl_mipi_csi: mipicsigrp-1 {
+		pinctrl_hog_sd3_vselect: hoggrp_sd3vselect {
 			fsl,pins = <
-				MX7D_PAD_GPIO1_IO01__ANATOP_24M_OUT 0x3
+				MX7D_PAD_GPIO1_IO13__SD3_VSELECT   0x59
 			>;
 		};
-	};
-};
-
-&usdhc1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_usdhc1>;
-	bus-width = <8>;
-	non-removable;
-	status = "okay";
-};
-
-&usdhc2 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
-	pinctrl-0 = <&pinctrl_usdhc2>;
-	pinctrl-1 = <&pinctrl_usdhc2_100mhz>;
-	pinctrl-2 = <&pinctrl_usdhc2_200mhz>;
-	cd-gpios = <&gpio5 9 0>;
-	wp-gpios = <&gpio5 10 0>;
-	keep-power-in-suspend;
-	enable-sdio-wakeup;
-	tuning-step = <2>;
-	vmmc-supply = <&reg_sd2_vmmc>;
-	status = "okay";
-};
-
-&usdhc3 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
-	pinctrl-0 = <&pinctrl_usdhc3>;
-	pinctrl-1 = <&pinctrl_usdhc3_100mhz>;
-	pinctrl-2 = <&pinctrl_usdhc3_200mhz>;
-	cd-gpios = <&gpio1 14 0>;
-	wp-gpios = <&gpio1 15 0>;
-	keep-power-in-suspend;
-	enable-sdio-wakeup;
-	tuning-step = <2>;
-	vmmc-supply = <&reg_sd3_vmmc>;
-	status = "okay";
-};
 
-&iomuxc {
-	csi1 {
 		pinctrl_csi: csigrp-1 {
 			fsl,pins = <
 				MX7D_PAD_LCD_DATA04__CSI_VSYNC 	0x0F
@@ -321,50 +395,21 @@
 				MX7D_PAD_LCD_DATA03__GPIO3_IO8 	0x0F
 			>;
 		};
-	};
 
-	lcdif {
-		pinctrl_lcdif_dat: lcdifdatgrp {
+		pinctrl_ecspi1_cs_1: ecspi1_cs_grp-1 {
 			fsl,pins = <
-				MX7D_PAD_LCD_DATA00__LCD_DATA0	0x79
-				MX7D_PAD_LCD_DATA01__LCD_DATA1	0x79
-				MX7D_PAD_LCD_DATA02__LCD_DATA2	0x79
-				MX7D_PAD_LCD_DATA03__LCD_DATA3	0x79
-				MX7D_PAD_LCD_DATA04__LCD_DATA4	0x79
-				MX7D_PAD_LCD_DATA05__LCD_DATA5	0x79
-				MX7D_PAD_LCD_DATA06__LCD_DATA6	0x79
-				MX7D_PAD_LCD_DATA07__LCD_DATA7	0x79
-				MX7D_PAD_LCD_DATA08__LCD_DATA8	0x79
-				MX7D_PAD_LCD_DATA09__LCD_DATA9	0x79
-				MX7D_PAD_LCD_DATA10__LCD_DATA10	0x79
-				MX7D_PAD_LCD_DATA11__LCD_DATA11	0x79
-				MX7D_PAD_LCD_DATA12__LCD_DATA12	0x79
-				MX7D_PAD_LCD_DATA13__LCD_DATA13	0x79
-				MX7D_PAD_LCD_DATA14__LCD_DATA14	0x79
-				MX7D_PAD_LCD_DATA15__LCD_DATA15	0x79
-				MX7D_PAD_LCD_DATA16__LCD_DATA16	0x79
-				MX7D_PAD_LCD_DATA17__LCD_DATA17	0x79
-				MX7D_PAD_LCD_DATA18__LCD_DATA18	0x79
-				MX7D_PAD_LCD_DATA19__LCD_DATA19	0x79
-				MX7D_PAD_LCD_DATA20__LCD_DATA20	0x79
-				MX7D_PAD_LCD_DATA21__LCD_DATA21	0x79
-				MX7D_PAD_LCD_DATA22__LCD_DATA22	0x79
-				MX7D_PAD_LCD_DATA23__LCD_DATA23	0x79
+				MX7D_PAD_ECSPI1_SS0__GPIO4_IO19 	0x2
 			>;
 		};
 
-		pinctrl_lcdif_ctrl: lcdifctrlgrp {
+		pinctrl_ecspi1_1: ecspi1grp-1 {
 			fsl,pins = <
-				MX7D_PAD_LCD_CLK__LCD_CLK	0x79
-				MX7D_PAD_LCD_ENABLE__LCD_ENABLE	0x79
-				MX7D_PAD_LCD_VSYNC__LCD_VSYNC	0x79
-				MX7D_PAD_LCD_HSYNC__LCD_HSYNC	0x79
+				MX7D_PAD_ECSPI1_SCLK__ECSPI1_SCLK 	0x2
+				MX7D_PAD_ECSPI1_MOSI__ECSPI1_MOSI 	0x2
+				MX7D_PAD_ECSPI1_MISO__ECSPI1_MISO 	0x2
 			>;
 		};
-	};
-
 
-	enet1 {
 		pinctrl_enet1: enet1grp {
 			fsl,pins = <
 				MX7D_PAD_GPIO1_IO10__ENET1_MDIO 0x3
@@ -383,9 +428,7 @@
 				MX7D_PAD_ENET1_RGMII_RX_CTL__ENET1_RGMII_RX_CTL 0x1
 			>;
 		};
-	};
 
-	enet2 {
 		pinctrl_enet2: enet2grp {
 			fsl,pins = <
 				MX7D_PAD_EPDC_GDSP__ENET2_RGMII_TXC	0x1
@@ -402,18 +445,14 @@
 				MX7D_PAD_EPDC_SDCE0__ENET2_RGMII_RX_CTL	0x1
 			>;
 		};
-	};
 
-	gpio_keys {
 		pinctrl_gpio_keys: gpio_keysgrp {
 			fsl,pins = <
 				MX7D_PAD_GPIO1_IO14__GPIO1_IO14 0x32
 				MX7D_PAD_GPIO1_IO15__GPIO1_IO15 0x32
 			>;
 		};
-	};
 
-	i2c1 {
 		pinctrl_i2c1_1: i2c1grp-1 {
 			fsl,pins = <
 				MX7D_PAD_I2C1_SDA__I2C1_SDA          0x4000007f
@@ -421,18 +460,13 @@
 			>;
 		};
 
-	};
-
-	i2c2 {
 		pinctrl_i2c2_1: i2c2grp-1 {
 			fsl,pins = <
 				MX7D_PAD_I2C2_SDA__I2C2_SDA          0x4000007f
 				MX7D_PAD_I2C2_SCL__I2C2_SCL          0x4000007f
 			>;
 		};
-	};
 
-	i2c3 {
 		pinctrl_i2c3_1: i2c3grp-1 {
 			fsl,pins = <
 				MX7D_PAD_I2C3_SDA__I2C3_SDA	     0x4000007f
@@ -440,18 +474,73 @@
 			>;
 		};
 
-	};
-
-	i2c4 {
 		pinctrl_i2c4_1: i2c4grp-1 {
 			fsl,pins = <
 				MX7D_PAD_I2C4_SDA__I2C4_SDA          0x4000007f
 				MX7D_PAD_I2C4_SCL__I2C4_SCL          0x4000007f
 			>;
 		};
-	};
 
-	uart1 {
+		pinctrl_lcdif_dat: lcdifdatgrp {
+			fsl,pins = <
+				MX7D_PAD_LCD_DATA00__LCD_DATA0	0x79
+				MX7D_PAD_LCD_DATA01__LCD_DATA1	0x79
+				MX7D_PAD_LCD_DATA02__LCD_DATA2	0x79
+				MX7D_PAD_LCD_DATA03__LCD_DATA3	0x79
+				MX7D_PAD_LCD_DATA04__LCD_DATA4	0x79
+				MX7D_PAD_LCD_DATA05__LCD_DATA5	0x79
+				MX7D_PAD_LCD_DATA06__LCD_DATA6	0x79
+				MX7D_PAD_LCD_DATA07__LCD_DATA7	0x79
+				MX7D_PAD_LCD_DATA08__LCD_DATA8	0x79
+				MX7D_PAD_LCD_DATA09__LCD_DATA9	0x79
+				MX7D_PAD_LCD_DATA10__LCD_DATA10	0x79
+				MX7D_PAD_LCD_DATA11__LCD_DATA11	0x79
+				MX7D_PAD_LCD_DATA12__LCD_DATA12	0x79
+				MX7D_PAD_LCD_DATA13__LCD_DATA13	0x79
+				MX7D_PAD_LCD_DATA14__LCD_DATA14	0x79
+				MX7D_PAD_LCD_DATA15__LCD_DATA15	0x79
+				MX7D_PAD_LCD_DATA16__LCD_DATA16	0x79
+				MX7D_PAD_LCD_DATA17__LCD_DATA17	0x79
+				MX7D_PAD_LCD_DATA18__LCD_DATA18	0x79
+				MX7D_PAD_LCD_DATA19__LCD_DATA19	0x79
+				MX7D_PAD_LCD_DATA20__LCD_DATA20	0x79
+				MX7D_PAD_LCD_DATA21__LCD_DATA21	0x79
+				MX7D_PAD_LCD_DATA22__LCD_DATA22	0x79
+				MX7D_PAD_LCD_DATA23__LCD_DATA23	0x79
+			>;
+		};
+
+		pinctrl_lcdif_ctrl: lcdifctrlgrp {
+			fsl,pins = <
+				MX7D_PAD_LCD_CLK__LCD_CLK	0x79
+				MX7D_PAD_LCD_ENABLE__LCD_ENABLE	0x79
+				MX7D_PAD_LCD_VSYNC__LCD_VSYNC	0x79
+				MX7D_PAD_LCD_HSYNC__LCD_HSYNC	0x79
+			>;
+		};
+
+		pinctrl_sai1: sai1grp {
+			fsl,pins = <
+				MX7D_PAD_SAI1_MCLK__SAI1_MCLK           0x1f
+				MX7D_PAD_SAI1_TX_BCLK__SAI1_TX_BCLK     0x1f
+				MX7D_PAD_SAI1_TX_SYNC__SAI1_TX_SYNC     0x1f
+				MX7D_PAD_SAI1_RX_SYNC__SAI1_RX_SYNC     0x1f
+				MX7D_PAD_SAI1_RX_DATA__SAI1_RX_DATA0    0x1f
+				MX7D_PAD_SAI1_TX_DATA__SAI1_TX_DATA0    0
+			>;
+		};
+
+		pinctrl_sai2: sai2grp {
+			fsl,pins = <
+				MX7D_PAD_SAI1_MCLK__SAI1_MCLK           0x1f
+				MX7D_PAD_SAI2_TX_BCLK__SAI2_TX_BCLK     0x1f
+				MX7D_PAD_SAI2_TX_SYNC__SAI2_TX_SYNC     0x1f
+				MX7D_PAD_SAI2_RX_DATA__SAI2_RX_DATA0    0x1f
+				MX7D_PAD_SAI2_TX_DATA__SAI2_TX_DATA0    0
+				MX7D_PAD_GPIO1_IO13__GPIO1_IO13		0x59
+			>;
+		};
+
 		pinctrl_uart1_1: uart1grp-1 {
 			fsl,pins = <
 				MX7D_PAD_UART1_TX_DATA__UART1_DCE_TX 0x79
@@ -459,9 +548,6 @@
 			>;
 		};
 
-	};
-
-	uart3 {
 		pinctrl_uart3_1: uart3grp-1 {
 			fsl,pins = <
 				MX7D_PAD_UART3_TX_DATA__UART3_DCE_TX	0x79
@@ -479,9 +565,7 @@
 				MX7D_PAD_UART3_CTS_B__UART3_DTE_RTS	0x79
 			>;
 		};
-	};
 
-	usdhc1 {
 		pinctrl_usdhc1: usdhc1grp {
 			fsl,pins = <
 				MX7D_PAD_SD1_CMD__SD1_CMD       0x59
@@ -496,9 +580,7 @@
 				MX7D_PAD_ECSPI2_SS0__SD1_DATA7  0x59
 			>;
 		};
-	};
 
-	usdhc2 {
 		pinctrl_usdhc2: usdhc2grp {
 			fsl,pins = <
 				MX7D_PAD_SD2_CMD__SD2_CMD     0x59
@@ -531,9 +613,7 @@
 				MX7D_PAD_SD2_DATA3__SD2_DATA3 0x5b
 			>;
 		};
-	};
 
-	usdhc3 {
 		pinctrl_usdhc3: usdhc3grp {
 			fsl,pins = <
 				MX7D_PAD_SD3_CMD__SD3_CMD     0x59
@@ -579,46 +659,56 @@
 			>;
 		};
 	};
+};
 
-	sai1 {
-		pinctrl_sai1: sai1grp {
+&iomuxc_lpsr {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog_2>;
+
+	imx7d-19x19-ddr3-arm2 {
+		pinctrl_hog_2: hoggrp-2 {
 			fsl,pins = <
-				MX7D_PAD_SAI1_MCLK__SAI1_MCLK           0x1f
-				MX7D_PAD_SAI1_TX_BCLK__SAI1_TX_BCLK     0x1f
-				MX7D_PAD_SAI1_TX_SYNC__SAI1_TX_SYNC     0x1f
-				MX7D_PAD_SAI1_RX_SYNC__SAI1_RX_SYNC     0x1f
-				MX7D_PAD_SAI1_RX_DATA__SAI1_RX_DATA0    0x1f
-				MX7D_PAD_SAI1_TX_DATA__SAI1_TX_DATA0    0
+				MX7D_PAD_GPIO1_IO03__GPIO1_IO3 0x80000000
+				MX7D_PAD_GPIO1_IO07__GPIO1_IO7 0x80000000
 			>;
 		};
-	};
 
-    sai2 {
-		pinctrl_sai2: sai2grp {
+		pinctrl_mipi_csi: mipicsigrp-1 {
 			fsl,pins = <
-				MX7D_PAD_SAI1_MCLK__SAI1_MCLK           0x1f
-				MX7D_PAD_SAI2_TX_BCLK__SAI2_TX_BCLK     0x1f
-				MX7D_PAD_SAI2_TX_SYNC__SAI2_TX_SYNC     0x1f
-				MX7D_PAD_SAI2_RX_DATA__SAI2_RX_DATA0    0x1f
-				MX7D_PAD_SAI2_TX_DATA__SAI2_TX_DATA0    0
-				MX7D_PAD_GPIO1_IO13__GPIO1_IO13		0x59
+				MX7D_PAD_GPIO1_IO01__ANATOP_24M_OUT 0x3
 			>;
 		};
 	};
+};
 
-	ecspi1 {
-		pinctrl_ecspi1_cs_1: ecspi1_cs_grp-1 {
-			fsl,pins = <
-				MX7D_PAD_ECSPI1_SS0__GPIO4_IO19 	0x2
-			>;
-		};
+&lcdif {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lcdif_dat
+		     &pinctrl_lcdif_ctrl>;
+	display = <&display0>;
+	status = "okay";
 
-		pinctrl_ecspi1_1: ecspi1grp-1 {
-			fsl,pins = <
-				MX7D_PAD_ECSPI1_SCLK__ECSPI1_SCLK 	0x2
-				MX7D_PAD_ECSPI1_MOSI__ECSPI1_MOSI 	0x2
-				MX7D_PAD_ECSPI1_MISO__ECSPI1_MISO 	0x2
-			>;
+	display0: display {
+		bits-per-pixel = <16>;
+		bus-width = <24>;
+
+		display-timings {
+			native-mode = <&timing0>;
+			timing0: timing0 {
+				clock-frequency = <33500000>;
+				hactive = <800>;
+				vactive = <480>;
+				hback-porch = <89>;
+				hfront-porch = <164>;
+				vback-porch = <23>;
+				vfront-porch = <10>;
+				hsync-len = <10>;
+				vsync-len = <10>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				de-active = <1>;
+				pixelclk-active = <0>;
+			};
 		};
 	};
 };
@@ -635,194 +725,62 @@
 	status = "disabled";
 };
 
-&epxp {
-        status = "okay";
-};
-
-&i2c1 {
-	clock-frequency = <100000>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c1_1>;
+&sdma {
 	status = "okay";
-
-	pmic: pfuze3000@08 {
-		compatible = "fsl,pfuze3000";
-		reg = <0x08>;
-
-		regulators {
-			sw1a_reg: sw1a {
-				regulator-min-microvolt = <700000>;
-				regulator-max-microvolt = <1475000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-			/* use sw1c_reg to align with pfuze100/pfuze200 */
-			sw1c_reg: sw1b {
-				regulator-min-microvolt = <700000>;
-				regulator-max-microvolt = <1475000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-
-			sw2_reg: sw2 {
-				regulator-min-microvolt = <1500000>;
-				regulator-max-microvolt = <1850000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw3a_reg: sw3 {
-				regulator-min-microvolt = <900000>;
-				regulator-max-microvolt = <1650000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			swbst_reg: swbst {
-				regulator-min-microvolt = <5000000>;
-				regulator-max-microvolt = <5150000>;
-			};
-
-			snvs_reg: vsnvs {
-				regulator-min-microvolt = <1000000>;
-				regulator-max-microvolt = <3000000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vref_reg: vrefddr {
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vgen1_reg: vldo1 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen2_reg: vldo2 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <1550000>;
-			};
-
-			vgen3_reg: vccsd {
-				regulator-min-microvolt = <2850000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen4_reg: v33 {
-				regulator-min-microvolt = <2850000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen5_reg: vldo3 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen6_reg: vldo4 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-		};
-	};
 };
 
-&sdma {
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1_1>;
+	assigned-clocks = <&clks IMX7D_UART1_ROOT_SRC>;
+	assigned-clock-parents = <&clks IMX7D_OSC_24M_CLK>;
 	status = "okay";
 };
 
-&i2c2 {
-	clock-frequency = <100000>;
+&uart3 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c2_1>;
+	pinctrl-0 = <&pinctrl_uart3_1>;
+	fsl,uart-has-rtscts;
+	assigned-clocks = <&clks IMX7D_UART3_ROOT_SRC>;
+	assigned-clock-parents = <&clks IMX7D_PLL_SYS_MAIN_240M_CLK>;
 	status = "okay";
+	/* for DTE mode, add below change */
+	/* fsl,dte-mode;*/
+	pinctrl-0 = <&pinctrl_uart3dte_1>;
 };
 
-&i2c3 {
-	clock-frequency = <100000>;
+&usdhc1 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c3_1>;
+	pinctrl-0 = <&pinctrl_usdhc1>;
+	bus-width = <8>;
+	non-removable;
 	status = "okay";
-
-	max7322: gpio@68 {
-		compatible = "maxim,max7322";
-		reg = <0x68>;
-		gpio-controller;
-		#gpio-cells = <2>;
-		resets = <&max7322_reset>;
-	};
-	codec: wm8958@1a {
-		compatible = "wlf,wm8958";
-		reg = <0x1a>;
-		clocks = <&clks IMX7D_AUDIO_MCLK_ROOT_CLK>,
-		       <&clks IMX7D_CLK_DUMMY>;
-		clock-names = "mclk1", "mclk2";
-
-		DBVDD1-supply = <&reg_aud_1v8>;
-		DBVDD2-supply = <&reg_aud_1v8>;
-		DBVDD3-supply = <&reg_aud_3v3>;
-		AVDD2-supply = <&reg_aud_1v8>;
-		CPVDD-supply = <&reg_aud_1v8>;
-		SPKVDD1-supply = <&reg_coedc_5v>;
-		SPKVDD2-supply = <&reg_coedc_5v>;
-
-		wlf,ldo1ena;
-		wlf,ldo2ena;
-	};
 };
 
-&ecspi1 {
-	    fsl,spi-num-chipselects = <1>;
-		cs-gpios = <&gpio4 19 0>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_ecspi1_1 &pinctrl_ecspi1_cs_1>;
-		status = "okay";
-
-	    flash: at45@0 {
-	        #address-cells = <1>;
-	        #size-cells = <1>;
-	        compatible = "atmel,at45";
-	        spi-max-frequency = <20000000>;
-	        reg = <0>;
-	    };
+&usdhc2 {
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc2>;
+	pinctrl-1 = <&pinctrl_usdhc2_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc2_200mhz>;
+	cd-gpios = <&gpio5 9 0>;
+	wp-gpios = <&gpio5 10 0>;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	tuning-step = <2>;
+	vmmc-supply = <&reg_sd2_vmmc>;
+	status = "okay";
 };
 
-&lcdif {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lcdif_dat
-		     &pinctrl_lcdif_ctrl>;
-	display = <&display0>;
+&usdhc3 {
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc3>;
+	pinctrl-1 = <&pinctrl_usdhc3_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc3_200mhz>;
+	cd-gpios = <&gpio1 14 0>;
+	wp-gpios = <&gpio1 15 0>;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	tuning-step = <2>;
+	vmmc-supply = <&reg_sd3_vmmc>;
 	status = "okay";
-
-	display0: display {
-		bits-per-pixel = <16>;
-		bus-width = <24>;
-
-		display-timings {
-			native-mode = <&timing0>;
-			timing0: timing0 {
-				clock-frequency = <33500000>;
-				hactive = <800>;
-				vactive = <480>;
-				hback-porch = <89>;
-				hfront-porch = <164>;
-				vback-porch = <23>;
-				vfront-porch = <10>;
-				hsync-len = <10>;
-				vsync-len = <10>;
-				hsync-active = <0>;
-				vsync-active = <0>;
-				de-active = <1>;
-				pixelclk-active = <0>;
-			};
-		};
-	};
 };
diff --git a/arch/arm/boot/dts/imx7d-19x19-lpddr3-arm2.dts b/arch/arm/boot/dts/imx7d-19x19-lpddr3-arm2.dts
index 276b375..4ad40e2 100644
--- a/arch/arm/boot/dts/imx7d-19x19-lpddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-19x19-lpddr3-arm2.dts
@@ -90,26 +90,6 @@
 	arm-supply = <&sw1a_reg>;
 };
 
-&iomuxc {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hog_1>;
-
-	hog {
-		pinctrl_hog_1: hoggrp-1 {
-			fsl,pins = <
-				MX7D_PAD_I2C4_SCL__GPIO4_IO14	   0x80000000
-
-				MX7D_PAD_SD1_CD_B__GPIO5_IO0 		0x59
-				MX7D_PAD_SD1_WP__GPIO5_IO1 			0x59
-				MX7D_PAD_SD1_RESET_B__GPIO5_IO2 	0x59
-
-				MX7D_PAD_GPIO1_IO14__GPIO1_IO14    	0x59
-				MX7D_PAD_GPIO1_IO15__GPIO1_IO15    	0x59
-			>;
-		};
-	};
-};
-
 &gpmi {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_gpmi_nand_1>;
@@ -117,94 +97,133 @@
 	nand-on-flash-bbt;
 };
 
-&uart1 {
+&i2c1 {
+	clock-frequency = <100000>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart1_1>;
-	assigned-clocks = <&clks IMX7D_UART1_ROOT_SRC>;
-	assigned-clock-parents = <&clks IMX7D_OSC_24M_CLK>;
+	pinctrl-0 = <&pinctrl_i2c1_1>;
 	status = "okay";
-};
 
-&uart3 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart3_1>;
-	fsl,uart-has-rtscts;
-	assigned-clocks = <&clks IMX7D_UART3_ROOT_SRC>;
-	assigned-clock-parents = <&clks IMX7D_PLL_SYS_MAIN_240M_CLK>;
-	status = "okay";
-	/* for DTE mode, add below change */
-	/* fsl,dte-mode;*/
-	pinctrl-0 = <&pinctrl_uart3dte_1>;
-};
+	pmic: pfuze3000@08 {
+		compatible = "fsl,pfuze3000";
+		reg = <0x08>;
 
-&iomuxc_lpsr {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hog_2>;
+		regulators {
+			sw1a_reg: sw1a {
+				regulator-min-microvolt = <700000>;
+				regulator-max-microvolt = <1475000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+			/* use sw1c_reg to align with pfuze100/pfuze200 */
+			sw1c_reg: sw1b {
+				regulator-min-microvolt = <700000>;
+				regulator-max-microvolt = <1475000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
 
-	hog {
-		pinctrl_hog_2: hoggrp-2 {
-			fsl,pins = <
-				MX7D_PAD_GPIO1_IO03__GPIO1_IO3 0x14
-				MX7D_PAD_GPIO1_IO05__GPIO1_IO5 0x14
-				MX7D_PAD_GPIO1_IO07__GPIO1_IO7 0x14
-			>;
-		};
-	};
-};
+			sw2_reg: sw2 {
+				regulator-min-microvolt = <1500000>;
+				regulator-max-microvolt = <1850000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
 
-&usbotg1 {
-	vbus-supply = <&reg_usb_otg1_vbus>;
-	status = "okay";
-};
+			sw3a_reg: sw3 {
+				regulator-min-microvolt = <900000>;
+				regulator-max-microvolt = <1650000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
 
-&usbotg2 {
-	vbus-supply = <&reg_usb_otg2_vbus>;
-	status = "okay";
+			swbst_reg: swbst {
+				regulator-min-microvolt = <5000000>;
+				regulator-max-microvolt = <5150000>;
+			};
+
+			snvs_reg: vsnvs {
+				regulator-min-microvolt = <1000000>;
+				regulator-max-microvolt = <3000000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vref_reg: vrefddr {
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vgen1_reg: vldo1 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen2_reg: vldo2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+			};
+
+			vgen3_reg: vccsd {
+				regulator-min-microvolt = <2850000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen4_reg: v33 {
+				regulator-min-microvolt = <2850000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen5_reg: vldo3 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen6_reg: vldo4 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+		};
+	};
 };
 
-&usdhc1 {
+&i2c2 {
+	clock-frequency = <100000>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_usdhc1>;
-	cd-gpios = <&gpio5 0 0>;
-	wp-gpios = <&gpio5 1 0>;
-	no-1-8-v;
-	keep-power-in-suspend;
-	enable-sdio-wakeup;
-	vmmc-supply = <&reg_sd1_vmmc>;
+	pinctrl-0 = <&pinctrl_i2c2_1>;
 	status = "okay";
 };
 
-&weim {
+&iomuxc {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_weim_nor_1 &pinctrl_weim_cs0_1>;
-	#address-cells = <2>;
-	#size-cells = <1>;
-	ranges = <0 0 0x28000000 0x08000000>;
-	status = "okay";
+	pinctrl-0 = <&pinctrl_hog_1>;
 
-	nor@0,0 {
-		compatible = "cfi-flash";
-		reg = <0 0 0x08000000>;
-		#address-cells = <1>;
-		#size-cells = <1>;
-		bank-width = <2>;
-		fsl,weim-cs-timing = <0x00610081 0x00000001 0x1c022000
-				0x0000c000 0x1404a38e 0x00000000>;
-	};
-};
+	imx7d-19x19-lpddr3-arm2 {
+		pinctrl_hog_1: hoggrp-1 {
+			fsl,pins = <
+				MX7D_PAD_I2C4_SCL__GPIO4_IO14	   0x80000000
 
-&iomuxc {
+				MX7D_PAD_SD1_CD_B__GPIO5_IO0 		0x59
+				MX7D_PAD_SD1_WP__GPIO5_IO1 			0x59
+				MX7D_PAD_SD1_RESET_B__GPIO5_IO2 	0x59
+
+				MX7D_PAD_GPIO1_IO14__GPIO1_IO14    	0x59
+				MX7D_PAD_GPIO1_IO15__GPIO1_IO15    	0x59
+			>;
+		};
 
-	gpio_keys {
 		pinctrl_gpio_keys: gpio_keysgrp {
 			fsl,pins = <
 				MX7D_PAD_GPIO1_IO14__GPIO1_IO14 0x32
 				MX7D_PAD_GPIO1_IO15__GPIO1_IO15 0x32
 			>;
 		};
-	};
-
-	gpmi {
 
 		pinctrl_gpmi_nand_1: gpmi-nand-1 {
 			fsl,pins = <
@@ -226,9 +245,7 @@
 				MX7D_PAD_SD3_DATA7__NAND_DATA07		0x71
 			>;
 		};
-	};
 
-	i2c1 {
 		pinctrl_i2c1_1: i2c1grp-1 {
 			fsl,pins = <
 				MX7D_PAD_I2C1_SDA__I2C1_SDA          0x4000007f
@@ -236,18 +253,73 @@
 			>;
 		};
 
-	};
-
-	i2c2 {
 		pinctrl_i2c2_1: i2c2grp-1 {
 			fsl,pins = <
 				MX7D_PAD_I2C2_SDA__I2C2_SDA          0x4000007f
 				MX7D_PAD_I2C2_SCL__I2C2_SCL          0x4000007f
 			>;
 		};
-	};
 
-	uart1 {
+		pinctrl_weim_cs0_1: weim_cs0grp-1 {
+			fsl,pins = <
+				MX7D_PAD_EPDC_DATA10__EIM_CS0_B      0x71
+			>;
+		};
+
+		pinctrl_weim_nor_1: weim_norgrp-1 {
+			fsl,pins = <
+				MX7D_PAD_EPDC_DATA08__EIM_OE         0x71
+				MX7D_PAD_EPDC_DATA09__EIM_RW         0x71
+				MX7D_PAD_EPDC_DATA11__EIM_BCLK       0x71
+				MX7D_PAD_EPDC_DATA12__EIM_LBA_B      0x71
+				MX7D_PAD_EPDC_DATA13__EIM_WAIT       0x75
+				/* data */
+				MX7D_PAD_LCD_DATA00__EIM_DATA0       0x7d
+				MX7D_PAD_LCD_DATA01__EIM_DATA1       0x7d
+				MX7D_PAD_LCD_DATA02__EIM_DATA2       0x7d
+				MX7D_PAD_LCD_DATA03__EIM_DATA3       0x7d
+				MX7D_PAD_LCD_DATA04__EIM_DATA4       0x7d
+				MX7D_PAD_LCD_DATA05__EIM_DATA5       0x7d
+				MX7D_PAD_LCD_DATA06__EIM_DATA6       0x7d
+				MX7D_PAD_LCD_DATA07__EIM_DATA7       0x7d
+				MX7D_PAD_LCD_DATA08__EIM_DATA8       0x7d
+				MX7D_PAD_LCD_DATA09__EIM_DATA9       0x7d
+				MX7D_PAD_LCD_DATA10__EIM_DATA10      0x7d
+				MX7D_PAD_LCD_DATA11__EIM_DATA11      0x7d
+				MX7D_PAD_LCD_DATA12__EIM_DATA12      0x7d
+				MX7D_PAD_LCD_DATA13__EIM_DATA13      0x7d
+				MX7D_PAD_LCD_DATA14__EIM_DATA14      0x7d
+				MX7D_PAD_LCD_DATA15__EIM_DATA15      0x7d
+				/* address */
+				MX7D_PAD_EPDC_DATA00__EIM_AD0        0x71
+				MX7D_PAD_EPDC_DATA01__EIM_AD1        0x71
+				MX7D_PAD_EPDC_DATA02__EIM_AD2        0x71
+				MX7D_PAD_EPDC_DATA03__EIM_AD3        0x71
+				MX7D_PAD_EPDC_DATA04__EIM_AD4        0x71
+				MX7D_PAD_EPDC_DATA05__EIM_AD5        0x71
+				MX7D_PAD_EPDC_DATA06__EIM_AD6        0x71
+				MX7D_PAD_EPDC_DATA07__EIM_AD7        0x71
+				MX7D_PAD_EPDC_BDR1__EIM_AD8          0x71
+				MX7D_PAD_EPDC_PWR_COM__EIM_AD9       0x71
+				MX7D_PAD_EPDC_SDCLK__EIM_AD10        0x71
+				MX7D_PAD_EPDC_SDLE__EIM_AD11         0x71
+				MX7D_PAD_EPDC_SDOE__EIM_AD12         0x71
+				MX7D_PAD_EPDC_SDSHR__EIM_AD13        0x71
+				MX7D_PAD_EPDC_SDCE0__EIM_AD14        0x71
+				MX7D_PAD_EPDC_SDCE1__EIM_AD15        0x71
+				MX7D_PAD_EPDC_SDCE2__EIM_ADDR16      0x71
+				MX7D_PAD_EPDC_SDCE3__EIM_ADDR17      0x71
+				MX7D_PAD_EPDC_GDCLK__EIM_ADDR18      0x71
+				MX7D_PAD_EPDC_GDOE__EIM_ADDR19       0x71
+				MX7D_PAD_EPDC_GDRL__EIM_ADDR20       0x71
+				MX7D_PAD_EPDC_GDSP__EIM_ADDR21       0x71
+				MX7D_PAD_EPDC_BDR0__EIM_ADDR22       0x71
+				MX7D_PAD_LCD_DATA20__EIM_ADDR23      0x71
+				MX7D_PAD_LCD_DATA21__EIM_ADDR24      0x71
+				MX7D_PAD_LCD_DATA22__EIM_ADDR25      0x71
+			>;
+		};
+
 		pinctrl_uart1_1: uart1grp-1 {
 			fsl,pins = <
 				MX7D_PAD_UART1_TX_DATA__UART1_DCE_TX 0x79
@@ -255,9 +327,6 @@
 			>;
 		};
 
-	};
-
-	uart3 {
 		pinctrl_uart3_1: uart3grp-1 {
 			fsl,pins = <
 				MX7D_PAD_UART3_TX_DATA__UART3_DCE_TX	0x79
@@ -275,9 +344,7 @@
 				MX7D_PAD_UART3_CTS_B__UART3_DTE_RTS		0x79
 			>;
 		};
-	};
 
-	usdhc1 {
 		pinctrl_usdhc1: usdhc1grp {
 			fsl,pins = <
 				MX7D_PAD_SD1_CMD__SD1_CMD       0x59
@@ -311,65 +378,18 @@
 			>;
 		};
 	};
+};
 
-	weim {
-		pinctrl_weim_cs0_1: weim_cs0grp-1 {
-			fsl,pins = <
-				MX7D_PAD_EPDC_DATA10__EIM_CS0_B      0x71
-			>;
-		};
+&iomuxc_lpsr {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog_2>;
 
-		pinctrl_weim_nor_1: weim_norgrp-1 {
+	imx7d-19x19-lpddr3-arm2 {
+		pinctrl_hog_2: hoggrp-2 {
 			fsl,pins = <
-				MX7D_PAD_EPDC_DATA08__EIM_OE         0x71
-				MX7D_PAD_EPDC_DATA09__EIM_RW         0x71
-				MX7D_PAD_EPDC_DATA11__EIM_BCLK       0x71
-				MX7D_PAD_EPDC_DATA12__EIM_LBA_B      0x71
-				MX7D_PAD_EPDC_DATA13__EIM_WAIT       0x75
-				/* data */
-				MX7D_PAD_LCD_DATA00__EIM_DATA0       0x7d
-				MX7D_PAD_LCD_DATA01__EIM_DATA1       0x7d
-				MX7D_PAD_LCD_DATA02__EIM_DATA2       0x7d
-				MX7D_PAD_LCD_DATA03__EIM_DATA3       0x7d
-				MX7D_PAD_LCD_DATA04__EIM_DATA4       0x7d
-				MX7D_PAD_LCD_DATA05__EIM_DATA5       0x7d
-				MX7D_PAD_LCD_DATA06__EIM_DATA6       0x7d
-				MX7D_PAD_LCD_DATA07__EIM_DATA7       0x7d
-				MX7D_PAD_LCD_DATA08__EIM_DATA8       0x7d
-				MX7D_PAD_LCD_DATA09__EIM_DATA9       0x7d
-				MX7D_PAD_LCD_DATA10__EIM_DATA10      0x7d
-				MX7D_PAD_LCD_DATA11__EIM_DATA11      0x7d
-				MX7D_PAD_LCD_DATA12__EIM_DATA12      0x7d
-				MX7D_PAD_LCD_DATA13__EIM_DATA13      0x7d
-				MX7D_PAD_LCD_DATA14__EIM_DATA14      0x7d
-				MX7D_PAD_LCD_DATA15__EIM_DATA15      0x7d
-				/* address */
-				MX7D_PAD_EPDC_DATA00__EIM_AD0        0x71
-				MX7D_PAD_EPDC_DATA01__EIM_AD1        0x71
-				MX7D_PAD_EPDC_DATA02__EIM_AD2        0x71
-				MX7D_PAD_EPDC_DATA03__EIM_AD3        0x71
-				MX7D_PAD_EPDC_DATA04__EIM_AD4        0x71
-				MX7D_PAD_EPDC_DATA05__EIM_AD5        0x71
-				MX7D_PAD_EPDC_DATA06__EIM_AD6        0x71
-				MX7D_PAD_EPDC_DATA07__EIM_AD7        0x71
-				MX7D_PAD_EPDC_BDR1__EIM_AD8          0x71
-				MX7D_PAD_EPDC_PWR_COM__EIM_AD9       0x71
-				MX7D_PAD_EPDC_SDCLK__EIM_AD10        0x71
-				MX7D_PAD_EPDC_SDLE__EIM_AD11         0x71
-				MX7D_PAD_EPDC_SDOE__EIM_AD12         0x71
-				MX7D_PAD_EPDC_SDSHR__EIM_AD13        0x71
-				MX7D_PAD_EPDC_SDCE0__EIM_AD14        0x71
-				MX7D_PAD_EPDC_SDCE1__EIM_AD15        0x71
-				MX7D_PAD_EPDC_SDCE2__EIM_ADDR16      0x71
-				MX7D_PAD_EPDC_SDCE3__EIM_ADDR17      0x71
-				MX7D_PAD_EPDC_GDCLK__EIM_ADDR18      0x71
-				MX7D_PAD_EPDC_GDOE__EIM_ADDR19       0x71
-				MX7D_PAD_EPDC_GDRL__EIM_ADDR20       0x71
-				MX7D_PAD_EPDC_GDSP__EIM_ADDR21       0x71
-				MX7D_PAD_EPDC_BDR0__EIM_ADDR22       0x71
-				MX7D_PAD_LCD_DATA20__EIM_ADDR23      0x71
-				MX7D_PAD_LCD_DATA21__EIM_ADDR24      0x71
-				MX7D_PAD_LCD_DATA22__EIM_ADDR25      0x71
+				MX7D_PAD_GPIO1_IO03__GPIO1_IO3 0x14
+				MX7D_PAD_GPIO1_IO05__GPIO1_IO5 0x14
+				MX7D_PAD_GPIO1_IO07__GPIO1_IO7 0x14
 			>;
 		};
 	};
@@ -379,105 +399,63 @@
 	status = "okay";
 };
 
-&i2c1 {
-	clock-frequency = <100000>;
+&weim {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c1_1>;
+	pinctrl-0 = <&pinctrl_weim_nor_1 &pinctrl_weim_cs0_1>;
+	#address-cells = <2>;
+	#size-cells = <1>;
+	ranges = <0 0 0x28000000 0x08000000>;
 	status = "okay";
 
-	pmic: pfuze3000@08 {
-		compatible = "fsl,pfuze3000";
-		reg = <0x08>;
-
-		regulators {
-			sw1a_reg: sw1a {
-				regulator-min-microvolt = <700000>;
-				regulator-max-microvolt = <1475000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-			/* use sw1c_reg to align with pfuze100/pfuze200 */
-			sw1c_reg: sw1b {
-				regulator-min-microvolt = <700000>;
-				regulator-max-microvolt = <1475000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-
-			sw2_reg: sw2 {
-				regulator-min-microvolt = <1500000>;
-				regulator-max-microvolt = <1850000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw3a_reg: sw3 {
-				regulator-min-microvolt = <900000>;
-				regulator-max-microvolt = <1650000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			swbst_reg: swbst {
-				regulator-min-microvolt = <5000000>;
-				regulator-max-microvolt = <5150000>;
-			};
-
-			snvs_reg: vsnvs {
-				regulator-min-microvolt = <1000000>;
-				regulator-max-microvolt = <3000000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vref_reg: vrefddr {
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vgen1_reg: vldo1 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen2_reg: vldo2 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <1550000>;
-			};
+	nor@0,0 {
+		compatible = "cfi-flash";
+		reg = <0 0 0x08000000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		bank-width = <2>;
+		fsl,weim-cs-timing = <0x00610081 0x00000001 0x1c022000
+				0x0000c000 0x1404a38e 0x00000000>;
+	};
+};
 
-			vgen3_reg: vccsd {
-				regulator-min-microvolt = <2850000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1_1>;
+	assigned-clocks = <&clks IMX7D_UART1_ROOT_SRC>;
+	assigned-clock-parents = <&clks IMX7D_OSC_24M_CLK>;
+	status = "okay";
+};
 
-			vgen4_reg: v33 {
-				regulator-min-microvolt = <2850000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
+&uart3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart3_1>;
+	fsl,uart-has-rtscts;
+	assigned-clocks = <&clks IMX7D_UART3_ROOT_SRC>;
+	assigned-clock-parents = <&clks IMX7D_PLL_SYS_MAIN_240M_CLK>;
+	status = "okay";
+	/* for DTE mode, add below change */
+	/* fsl,dte-mode;*/
+	pinctrl-0 = <&pinctrl_uart3dte_1>;
+};
 
-			vgen5_reg: vldo3 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
+&usbotg1 {
+	vbus-supply = <&reg_usb_otg1_vbus>;
+	status = "okay";
+};
 
-			vgen6_reg: vldo4 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-		};
-	};
+&usbotg2 {
+	vbus-supply = <&reg_usb_otg2_vbus>;
+	status = "okay";
 };
 
-&i2c2 {
-	clock-frequency = <100000>;
+&usdhc1 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c2_1>;
+	pinctrl-0 = <&pinctrl_usdhc1>;
+	cd-gpios = <&gpio5 0 0>;
+	wp-gpios = <&gpio5 1 0>;
+	no-1-8-v;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	vmmc-supply = <&reg_sd1_vmmc>;
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx7d-sdb-epdc.dts b/arch/arm/boot/dts/imx7d-sdb-epdc.dts
index 983b4f8..9dcb9d7 100644
--- a/arch/arm/boot/dts/imx7d-sdb-epdc.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-epdc.dts
@@ -8,6 +8,10 @@
 
 #include "imx7d-sdb.dts"
 
+&epdc {
+        status = "okay";
+};
+
 &fec1 {
 	status = "disabled";
 };
@@ -16,10 +20,6 @@
 	status = "disabled";
 };
 
-&epdc {
-        status = "okay";
-};
-
 &max17135 {
         status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx7d-sdb-gpmi-weim.dts b/arch/arm/boot/dts/imx7d-sdb-gpmi-weim.dts
index 8b27ffe..2014f0a 100644
--- a/arch/arm/boot/dts/imx7d-sdb-gpmi-weim.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-gpmi-weim.dts
@@ -8,6 +8,10 @@
 
 #include "imx7d-sdb.dts"
 
+&gpmi{
+	status = "okay";
+};
+
 &sai1{
 	status = "disabled";
 };
@@ -19,7 +23,3 @@
 &uart5{
 	status = "disabled";
 };
-
-&gpmi{
-	status = "okay";
-};
diff --git a/arch/arm/boot/dts/imx7d-sdb-m4.dts b/arch/arm/boot/dts/imx7d-sdb-m4.dts
index e9a92c9..7b57979 100644
--- a/arch/arm/boot/dts/imx7d-sdb-m4.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-m4.dts
@@ -15,18 +15,29 @@
 	};
 };
 
-&flexcan2 {
+&adc1 {
 	status = "disabled";
 };
 
-&adc1 {
+&adc2 {
 	status = "disabled";
 };
 
-&adc2 {
+&flexcan2 {
 	status = "disabled";
 };
 
+&iomuxc {
+	imx7d-sdb {
+		pinctrl_uart2_1: uart2grp-1 {
+			fsl,pins = <
+				MX7D_PAD_UART2_RX_DATA__UART2_DCE_RX    0x79
+				MX7D_PAD_UART2_TX_DATA__UART2_DCE_TX    0x79
+			>;
+		};
+	};
+};
+
 &mcctest{
 	status = "okay";
 };
@@ -39,17 +50,6 @@
 	status = "okay";
 };
 
-&iomuxc {
-	uart2 {
-		pinctrl_uart2_1: uart2grp-1 {
-			fsl,pins = <
-				MX7D_PAD_UART2_RX_DATA__UART2_DCE_RX    0x79
-				MX7D_PAD_UART2_TX_DATA__UART2_DCE_TX    0x79
-			>;
-		};
-	};
-};
-
 &uart2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart2_1>;
diff --git a/arch/arm/boot/dts/imx7d-sdb-touch.dts b/arch/arm/boot/dts/imx7d-sdb-touch.dts
index 3f854e3..e282d14 100644
--- a/arch/arm/boot/dts/imx7d-sdb-touch.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-touch.dts
@@ -1,9 +1,5 @@
 #include "imx7d-sdb.dts"
 
-&sii902x {
-	status = "disabled";
-};
-
 &ecspi3 {
 	fsl,spi-num-chipselects = <1>;
 	pinctrl-names = "default";
@@ -30,3 +26,6 @@
 	};
 };
 
+&sii902x {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 2c15dc0..b4b3c9e 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -160,24 +160,6 @@
 	};
 };
 
-&cpu0 {
-	arm-supply = <&sw1a_reg>;
-};
-
-&epdc {
-        pinctrl-names = "default";
-        pinctrl-0 = <&pinctrl_epdc0>;
-        V3P3-supply = <&V3P3_reg>;
-        VCOM-supply = <&VCOM_reg>;
-        DISPLAY-supply = <&DISPLAY_reg>;
-	en-gpios = <&gpio_spi 5 0>;
-        status = "disabled";
-};
-
-&epxp {
-	status = "okay";
-};
-
 &adc1 {
 	vref-supply = <&reg_vref_1v8>;
 	status = "okay";
@@ -188,6 +170,15 @@
 	status = "okay";
 };
 
+&clks {
+	assigned-clocks = <&clks IMX7D_PLL_AUDIO_POST_DIV>;
+	assigned-clock-rates = <884736000>;
+};
+
+&cpu0 {
+	arm-supply = <&sw1a_reg>;
+};
+
 &csi1 {
 	csi-mux-mipi = <&gpr 0x14 4>;
 	status = "okay";
@@ -199,6 +190,19 @@
 	};
 };
 
+&epdc {
+        pinctrl-names = "default";
+        pinctrl-0 = <&pinctrl_epdc0>;
+        V3P3-supply = <&V3P3_reg>;
+        VCOM-supply = <&VCOM_reg>;
+        DISPLAY-supply = <&DISPLAY_reg>;
+	en-gpios = <&gpio_spi 5 0>;
+        status = "disabled";
+};
+
+&epxp {
+	status = "okay";
+};
 
 &fec1 {
 	pinctrl-names = "default";
@@ -513,46 +517,6 @@
 	};
 };
 
-&lcdif {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lcdif_dat
-		     &pinctrl_lcdif_ctrl>;
-	pinctrl-assert-gpios = <&gpio_spi 7 GPIO_ACTIVE_HIGH>;
-	display = <&display0>;
-	status = "okay";
-
-	display0: display {
-		bits-per-pixel = <16>;
-		bus-width = <24>;
-
-		display-timings {
-			native-mode = <&timing0>;
-			timing0: timing0 {
-			clock-frequency = <9200000>;
-			hactive = <480>;
-			vactive = <272>;
-			hfront-porch = <8>;
-			hback-porch = <4>;
-			hsync-len = <41>;
-			vback-porch = <2>;
-			vfront-porch = <4>;
-			vsync-len = <10>;
-
-			hsync-active = <0>;
-			vsync-active = <0>;
-			de-active = <1>;
-			pixelclk-active = <0>;
-			};
-		};
-	};
-};
-
-&pwm1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pwm1>;
-	status = "okay";
-};
-
 &iomuxc {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_hog_1>;
@@ -881,7 +845,6 @@
 			>;
 		};
 
-
 		pinctrl_usdhc3: usdhc3grp {
 			fsl,pins = <
 				MX7D_PAD_SD3_CMD__SD3_CMD	0x59
@@ -971,6 +934,7 @@
 				MX7D_PAD_GPIO1_IO00__WDOD1_WDOG_B 0x74
 			>;
 		};
+
 		pinctrl_pwm1: pwm1grp {
 			fsl,pins = <
 				MX7D_PAD_GPIO1_IO01__PWM1_OUT	0x110b0
@@ -979,6 +943,40 @@
 	};
 };
 
+&lcdif {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lcdif_dat
+		     &pinctrl_lcdif_ctrl>;
+	pinctrl-assert-gpios = <&gpio_spi 7 GPIO_ACTIVE_HIGH>;
+	display = <&display0>;
+	status = "okay";
+
+	display0: display {
+		bits-per-pixel = <16>;
+		bus-width = <24>;
+
+		display-timings {
+			native-mode = <&timing0>;
+			timing0: timing0 {
+			clock-frequency = <9200000>;
+			hactive = <480>;
+			vactive = <272>;
+			hfront-porch = <8>;
+			hback-porch = <4>;
+			hsync-len = <41>;
+			vback-porch = <2>;
+			vfront-porch = <4>;
+			vsync-len = <10>;
+
+			hsync-active = <0>;
+			vsync-active = <0>;
+			de-active = <1>;
+			pixelclk-active = <0>;
+			};
+		};
+	};
+};
+
 &pcie {
 	pinctrl-names = "default";
 	reset-gpio = <&gpio_spi 1 GPIO_ACTIVE_LOW>;
@@ -986,9 +984,10 @@
 	status = "okay";
 };
 
-&clks {
-	assigned-clocks = <&clks IMX7D_PLL_AUDIO_POST_DIV>;
-	assigned-clock-rates = <884736000>;
+&pwm1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm1>;
+	status = "okay";
 };
 
 &sai1 {
-- 
1.7.5.4

